// @strapi/design-system/dist/Divider/Divider.js
var c = Object.defineProperty, f = Object.defineProperties;
var l = Object.getOwnPropertyDescriptors;
var n = Object.getOwnPropertySymbols;
var i = Object.prototype.hasOwnProperty, a = Object.prototype.propertyIsEnumerable;
var p = (r, e, o) => e in r ? c(r, e, { enumerable: !0, configurable: !0, writable: !0, value: o }) : r[e] = o, m = (r, e) => {
  for (var o in e || (e = {}))
    i.call(e, o) && p(r, o, e[o]);
  if (n)
    for (var o of n(e))
      a.call(e, o) && p(r, o, e[o]);
  return r;
}, s = (r, e) => f(r, l(e));
var d = (r, e) => {
  var o = {};
  for (var t in r)
    i.call(r, t) && e.indexOf(t) < 0 && (o[t] = r[t]);
  if (r != null && n)
    for (var t of n(r))
      e.indexOf(t) < 0 && a.call(r, t) && (o[t] = r[t]);
  return o;
};
import b from "react";
import u from "prop-types";
import h from "styled-components";
import { Box as x } from "../Box/Box.js";
const y = h(x)`
  height: 1px;
  border: none;
  ${({ unsetMargin: r }) => r ? "margin: 0;" : ""}
`, g = (o) => {
  var t = o, { unsetMargin: r } = t, e = d(t, ["unsetMargin"]);
  return /* @__PURE__ */ b.createElement(y, s(m({}, e), {
    as: "hr",
    unsetMargin: r
  }));
};
g.defaultProps = {
  background: "neutral150",
  unsetMargin: !0
};
g.propTypes = {
  background: u.string,
  unsetMargin: u.bool
};
export {
  g as Divider
};
