// @strapi/design-system/dist/DateTimePicker/DateTimePickerProps.js
var s = Object.defineProperty;
var o = Object.getOwnPropertySymbols;
var d = Object.prototype.hasOwnProperty, c = Object.prototype.propertyIsEnumerable;
var l = (t, i, r) => i in t ? s(t, i, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[i] = r, a = (t, i) => {
  for (var r in i || (i = {}))
    d.call(i, r) && l(t, r, i[r]);
  if (o)
    for (var r of o(i))
      c.call(i, r) && l(t, r, i[r]);
  return t;
};
import p from "react";
import e from "prop-types";
const n = (t) => /* @__PURE__ */ p.createElement("div", a({}, t)), b = {
  ariaLabel: void 0,
  clearLabel: "clear",
  disabled: !1,
  error: void 0,
  hint: void 0,
  label: void 0,
  labelAction: void 0,
  onClear: void 0,
  required: !1,
  selectButtonTitle: "select",
  size: "M",
  step: 1,
  value: void 0
}, u = {
  ariaLabel: e.string,
  clearLabel: e.string,
  disabled: e.bool,
  error: e.oneOfType([e.string, e.bool]),
  hint: e.string,
  label: e.string,
  labelAction: e.element,
  name: e.string.isRequired,
  onChange: e.func.isRequired,
  onClear: e.func,
  required: e.bool,
  selectButtonTitle: e.string,
  size: e.oneOf(["S", "M"]),
  step: e.number,
  value: e.instanceOf(Date)
};
n.defaultProps = b;
n.propTypes = u;
export {
  n as DateTimePickerProps,
  b as dateTimePickerDefaultProps,
  u as dateTimePickerPropTypes
};
