// @strapi/design-system/dist/Pagination/components.js
var R = Object.defineProperty, T = Object.defineProperties;
var N = Object.getOwnPropertyDescriptors;
var m = Object.getOwnPropertySymbols;
var y = Object.prototype.hasOwnProperty, P = Object.prototype.propertyIsEnumerable;
var b = (e, r, t) => r in e ? R(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, l = (e, r) => {
  for (var t in r || (r = {}))
    y.call(r, t) && b(e, t, r[t]);
  if (m)
    for (var t of m(r))
      P.call(r, t) && b(e, t, r[t]);
  return e;
}, v = (e, r) => T(e, N(r));
var d = (e, r) => {
  var t = {};
  for (var o in e)
    y.call(e, o) && r.indexOf(o) < 0 && (t[o] = e[o]);
  if (e != null && m)
    for (var o of m(e))
      r.indexOf(o) < 0 && P.call(e, o) && (t[o] = e[o]);
  return t;
};
import i from "react";
import p from "styled-components";
import c from "prop-types";
import q from "@strapi/icons/ChevronLeft";
import W from "@strapi/icons/ChevronRight";
import { NavLink as C } from "react-router-dom";
import { VisuallyHidden as u } from "../VisuallyHidden/VisuallyHidden.js";
import { usePagination as f } from "./PaginationContext.js";
import { Typography as S } from "../Typography/Typography.js";
import { buttonFocusStyle as A } from "../themes/utils.js";
const E = p(S)`
  line-height: revert;
`, D = {
  active: !0
}, g = p(C).withConfig({
  shouldForwardProp: (e, r) => !D[e] && r(e)
})`
  padding: ${({ theme: e }) => e.spaces[3]};
  border-radius: ${({ theme: e }) => e.borderRadius};
  box-shadow: ${({ active: e, theme: r }) => e ? r.shadows.filterShadow : void 0};
  text-decoration: none;
  display: flex;

  ${A}
`, F = p(g)`
  color: ${({ theme: e, active: r }) => r ? e.colors.primary700 : e.colors.neutral800};
  background: ${({ theme: e, active: r }) => r ? e.colors.neutral0 : void 0};

  &:hover {
    box-shadow: ${({ theme: e }) => e.shadows.filterShadow};
  }
`, k = p(g)`
  font-size: 0.7rem;
  svg path {
    fill: ${(e) => e["aria-disabled"] ? e.theme.colors.neutral300 : e.theme.colors.neutral600};
  }

  &:focus,
  &:hover {
    svg path {
      fill: ${(e) => e["aria-disabled"] ? e.theme.colors.neutral300 : e.theme.colors.neutral700};
    }
  }

  ${(e) => e["aria-disabled"] ? `
  pointer-events: none;
    ` : void 0}
`, I = p(g)`
  color: ${({ theme: e }) => e.colors.neutral800};
`, L = (o) => {
  var a = o, { children: e, to: r } = a, t = d(a, ["children", "to"]);
  const { activePage: s } = f(), n = s === 1;
  return /* @__PURE__ */ i.createElement(k, l({
    to: n ? "#" : r,
    "aria-disabled": n,
    tabIndex: n ? -1 : void 0
  }, t), /* @__PURE__ */ i.createElement(u, null, e), /* @__PURE__ */ i.createElement(q, {
    "aria-hidden": !0
  }));
};
L.displayName = "PreviousLink";
const $ = (o) => {
  var a = o, { children: e, to: r } = a, t = d(a, ["children", "to"]);
  const { activePage: s, pageCount: n } = f(), h = s === n;
  return /* @__PURE__ */ i.createElement(k, l({
    to: h ? "#" : r,
    "aria-disabled": h,
    tabIndex: h ? -1 : void 0
  }, t), /* @__PURE__ */ i.createElement(u, null, e), /* @__PURE__ */ i.createElement(W, {
    "aria-hidden": !0
  }));
};
$.displayName = "NextLink";
const x = (o) => {
  var a = o, { number: e, children: r } = a, t = d(a, ["number", "children"]);
  const { activePage: s } = f(), n = s === e;
  return /* @__PURE__ */ i.createElement(F, v(l({}, t), {
    active: n
  }), /* @__PURE__ */ i.createElement(u, null, r), /* @__PURE__ */ i.createElement(E, {
    "aria-hidden": !0,
    variant: "pi",
    fontWeight: n ? "bold" : null
  }, e));
};
x.displayName = "PageLink";
const _ = (t) => {
  var o = t, { children: e } = o, r = d(o, ["children"]);
  return /* @__PURE__ */ i.createElement(I, v(l({}, r), {
    as: "div"
  }), /* @__PURE__ */ i.createElement(u, null, e), /* @__PURE__ */ i.createElement(E, {
    "aria-hidden": !0,
    variant: "pi"
  }, "\u2026"));
};
x.propTypes = {
  children: c.node.isRequired,
  number: c.number.isRequired
};
const w = {
  children: c.node.isRequired,
  to: c.string.isRequired
};
$.propTypes = w;
L.propTypes = w;
_.propTypes = {
  children: c.node.isRequired
};
export {
  _ as Dots,
  $ as NextLink,
  x as PageLink,
  L as PreviousLink
};
