// @strapi/design-system/dist/v2/Breadcrumbs/CrumbSimpleMenu.js
var d = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var m = Object.prototype.hasOwnProperty, a = Object.prototype.propertyIsEnumerable;
var p = (e, r, o) => r in e ? d(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, l = (e, r) => {
  for (var o in r || (r = {}))
    m.call(r, o) && p(e, o, r[o]);
  if (i)
    for (var o of i(r))
      a.call(r, o) && p(e, o, r[o]);
  return e;
};
var s = (e, r) => {
  var o = {};
  for (var t in e)
    m.call(e, t) && r.indexOf(t) < 0 && (o[t] = e[t]);
  if (e != null && i)
    for (var t of i(e))
      r.indexOf(t) < 0 && a.call(e, t) && (o[t] = e[t]);
  return o;
};
import n from "prop-types";
import u from "react";
import f from "styled-components";
import y from "@strapi/icons/CarretDown";
import { Button as S } from "../../Button/Button.js";
import { SimpleMenu as b } from "../SimpleMenu/SimpleMenu.js";
const $ = f(S)`
  padding: ${({ theme: e }) => `${e.spaces[1]} ${e.spaces[3]}`};

  :hover,
  :focus {
    background-color: ${({ theme: e }) => e.colors.neutral200};
  }
`, c = (o) => {
  var t = o, { children: e } = t, r = s(t, ["children"]);
  return /* @__PURE__ */ u.createElement(b, l({
    noBorder: !0,
    icon: /* @__PURE__ */ u.createElement(y, null),
    as: $,
    size: "S"
  }, r), e);
};
c.displayName = "CrumbSimpleMenu";
c.propTypes = {
  "aria-label": n.string.isRequired,
  children: n.oneOfType([n.arrayOf(n.node), n.node]).isRequired
};
export {
  c as CrumbSimpleMenu
};
