// @strapi/design-system/dist/Combobox/utils.js
import c from "compute-scroll-into-view";
import { KeyboardKeys as n } from "../helpers/keyboardKeys.js";
const t = {
  Close: "Close",
  CloseSelect: "CloseSelect",
  First: "First",
  Last: "Last",
  Next: "Next",
  Open: "Open",
  PageDown: "PageDown",
  PageUp: "PageUp",
  Previous: "Previous",
  Select: "Select",
  Space: "Space",
  Type: "Type"
}, f = {
  Close: "Close",
  First: "First",
  Last: "Last",
  Next: "Next",
  Open: "Open",
  Previous: "Previous",
  Select: "Select",
  UpLevel: "UpLevel"
};
function d(e = [], r = null, o = []) {
  const s = String(r != null ? r : "").toLowerCase();
  return s ? e.filter((i) => i.props.children.toString().toLowerCase().includes(s) && o.indexOf(i) < 0) : e;
}
function C(e, r) {
  if (!r && e === n.DOWN)
    return t.Open;
  if (e === n.DOWN)
    return t.Next;
  if (e === n.UP)
    return t.Previous;
  if (e === n.HOME)
    return t.First;
  if (e === n.END)
    return t.Last;
  if (e === n.ESCAPE)
    return t.Close;
  if (e === n.ENTER)
    return t.CloseSelect;
  if (e === n.BACKSPACE || e === n.CLEAR || e.length === 1)
    return t.Type;
}
function S(e, r, o) {
  switch (o) {
    case t.First:
      return 0;
    case t.Last:
      return r;
    case t.Previous:
      return Math.max(0, e - 1);
    case t.Next:
      return Math.min(r, e + 1);
    default:
      return e;
  }
}
function P(e) {
  c(e, {
    scrollMode: "if-needed",
    block: "nearest",
    inline: "nearest"
  }).forEach(({ el: o, top: s, left: i }) => {
    o.scrollTop = s, o.scrollLeft = i;
  });
}
export {
  t as MenuActions,
  f as TreeActions,
  d as filterOptions,
  C as getActionFromKey,
  S as getUpdatedIndex,
  P as maintainScrollVisibility
};
