// @strapi/design-system/dist/Avatar/Avatar.cjs
"use strict";Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const d=require("react"),f=require("styled-components"),v=require("prop-types"),t=require("./constants.cjs"),g=require("../Typography/Typography.cjs"),h=require("../Flex/Flex.cjs"),l=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},a=l(d),o=l(f),r=l(v),y=o.default.img`
  border-radius: 50%;
  object-fit: cover;
  display: block;
  position: relative;
`,m=o.default.div`
  position: relative;
  width: ${t.avatarSize/16}rem;
  height: ${t.avatarSize/16}rem;
  z-index: ${({hovering:e})=>e?1:void 0};
`,b=o.default.img`
  border-radius: 50%;
  object-fit: cover;
  position: absolute;
  transform: translate(-${(t.previewSize-t.avatarSize)/2}px, -100%);
  margin-top: -${({theme:e})=>e.spaces[1]};
`,S=o.default.div`
  z-index: 1;
  border-radius: 30%;
  position: absolute;
  width: ${t.avatarSize/16}rem;
  height: ${t.avatarSize/16}rem;
  background: ${({theme:e})=>e.colors.neutral0};
  opacity: 0.4;
`,s=({src:e,alt:c,preview:i})=>{const[n,u]=d.useState(!1);return a.default.createElement("span",null,i&&n?a.default.createElement(b,{"aria-hidden":!0,alt:"",width:`${t.previewSize}px`,height:`${t.previewSize}px`,src:i===!0?e:i}):null,a.default.createElement(m,{hovering:i&&n,onMouseEnter:()=>u(!0),onMouseLeave:()=>u(!1)},i&&n?a.default.createElement(S,null):null,a.default.createElement(y,{src:e,alt:c,width:`${t.avatarSize}px`,height:`${t.avatarSize}px`})))},x=o.default(h.Flex)`
  span {
    line-height: 0;
  }
`,p=({children:e})=>a.default.createElement(x,{borderRadius:"50%",width:`${t.avatarSize}px`,height:`${t.avatarSize}px`,background:"primary600",justifyContent:"center"},a.default.createElement(g.Typography,{fontWeight:"bold",textColor:"buttonNeutral0",fontSize:0,textTransform:"uppercase"},e));p.propTypes={children:r.default.node.isRequired};s.defaultProps={alt:void 0,preview:void 0};s.propTypes={alt:r.default.string,preview:r.default.oneOfType([r.default.string,r.default.bool]),src:r.default.string.isRequired};exports.Avatar=s;exports.Initials=p;
