// @strapi/design-system/dist/BaseLink/BaseLink.js
var y = Object.defineProperty;
var s = Object.getOwnPropertySymbols;
var p = Object.prototype.hasOwnProperty, d = Object.prototype.propertyIsEnumerable;
var i = (e, o, r) => o in e ? y(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, m = (e, o) => {
  for (var r in o || (o = {}))
    p.call(o, r) && i(e, r, o[r]);
  if (s)
    for (var r of s(o))
      d.call(o, r) && i(e, r, o[r]);
  return e;
};
var g = (e, o) => {
  var r = {};
  for (var t in e)
    p.call(e, t) && o.indexOf(t) < 0 && (r[t] = e[t]);
  if (e != null && s)
    for (var t of s(e))
      o.indexOf(t) < 0 && d.call(e, t) && (r[t] = e[t]);
  return r;
};
import c from "react";
import a from "prop-types";
import _ from "styled-components";
const h = _.a`
  cursor: pointer;
`, l = c.forwardRef((k, u) => {
  var f = k, { href: e, rel: o, target: r, disabled: t, isExternal: n } = f, b = g(f, ["href", "rel", "target", "disabled", "isExternal"]);
  return /* @__PURE__ */ c.createElement(h, m({
    ref: u,
    target: n ? "_blank" : r,
    rel: n ? o : void 0,
    href: t ? "#" : e,
    disabled: t
  }, b));
});
l.displayName = "BaseLink";
l.defaultProps = {
  disabled: !1,
  href: void 0,
  isExternal: !1,
  rel: "noreferrer noopener",
  target: "_self"
};
l.propTypes = {
  disabled: a.bool,
  href: a.string,
  isExternal: a.bool,
  rel: a.string,
  target: a.string
};
export {
  l as BaseLink
};
