// @strapi/design-system/dist/v2/index.js
import { Breadcrumbs as e } from "./Breadcrumbs/Breadcrumbs.js";
import { Crumb as m } from "./Breadcrumbs/Crumb.js";
import { CrumbLink as p } from "./Breadcrumbs/CrumbLink.js";
import { CrumbSimpleMenu as f } from "./Breadcrumbs/CrumbSimpleMenu.js";
import { Link as i } from "./Link/Link.js";
import { LinkButton as v } from "./LinkButton/LinkButton.js";
import { MainNav as S } from "./MainNav/MainNav.js";
import { NavBrand as k } from "./MainNav/NavBrand.js";
import { NavLink as s } from "./MainNav/NavLink.js";
import { NavSection as d } from "./MainNav/NavSection.js";
import { NavSections as M } from "./MainNav/NavSections.js";
import { NavUser as P } from "./MainNav/NavUser.js";
import { NavCondense as l } from "./MainNav/NavCondense.js";
import { Pagination as D } from "./Pagination/Pagination.js";
import { Dots as I, NextLink as U, PageLink as W, PreviousLink as j } from "./Pagination/components.js";
import { MenuItem as w, SimpleMenu as y } from "./SimpleMenu/SimpleMenu.js";
import { SubNav as A, subNavWidth as E } from "./SubNav/SubNav.js";
import { SubNavHeader as G } from "./SubNav/SubNavHeader.js";
import { SubNavLink as K } from "./SubNav/SubNavLink.js";
import { SubNavLinkSection as Q } from "./SubNav/SubNavLinkSection.js";
import { SubNavSection as T } from "./SubNav/SubNavSection.js";
import { SubNavSections as X } from "./SubNav/SubNavSections.js";
export {
  e as Breadcrumbs,
  m as Crumb,
  p as CrumbLink,
  f as CrumbSimpleMenu,
  I as Dots,
  i as Link,
  v as LinkButton,
  S as MainNav,
  w as MenuItem,
  k as NavBrand,
  l as NavCondense,
  s as NavLink,
  d as NavSection,
  M as NavSections,
  P as NavUser,
  U as NextLink,
  W as PageLink,
  D as Pagination,
  j as PreviousLink,
  y as SimpleMenu,
  A as SubNav,
  G as SubNavHeader,
  K as SubNavLink,
  Q as SubNavLinkSection,
  T as SubNavSection,
  X as SubNavSections,
  E as subNavWidth
};
