// @strapi/design-system/dist/RawTable/RawTable.js
var q = Object.defineProperty;
var f = Object.getOwnPropertySymbols;
var p = Object.prototype.hasOwnProperty, E = Object.prototype.propertyIsEnumerable;
var y = (t, a, r) => a in t ? q(t, a, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[a] = r, P = (t, a) => {
  for (var r in a || (a = {}))
    p.call(a, r) && y(t, r, a[r]);
  if (f)
    for (var r of f(a))
      E.call(a, r) && y(t, r, a[r]);
  return t;
};
var T = (t, a) => {
  var r = {};
  for (var c in t)
    p.call(t, c) && a.indexOf(c) < 0 && (r[c] = t[c]);
  if (t != null && f)
    for (var c of f(t))
      a.indexOf(c) < 0 && E.call(t, c) && (r[c] = t[c]);
  return r;
};
import K, { useRef as R, useState as d, useCallback as w, useEffect as A } from "react";
import u from "prop-types";
import { RawTableContext as F } from "./RawTableContext.js";
import { KeyboardKeys as s } from "../helpers/keyboardKeys.js";
import { focusFocusable as H } from "./focusFocusable.js";
const _ = (U) => {
  var b = U, { colCount: t, rowCount: a, jumpStep: r, initialCol: c, initialRow: G } = b, N = T(b, ["colCount", "rowCount", "jumpStep", "initialCol", "initialRow"]);
  const m = R(null), o = R(!1), [D, n] = d(G), [k, i] = d(c), O = w(({ colIndex: l, rowIndex: e }) => {
    i(l), n(e);
  }, []);
  A(() => {
    o.current && H(m.current), o.current || (o.current = !0);
  }, [k, D]);
  const h = (l) => {
    switch (l.key) {
      case s.RIGHT: {
        l.preventDefault(), i((e) => e < t - 1 ? e + 1 : e);
        break;
      }
      case s.LEFT: {
        l.preventDefault(), i((e) => e > 0 ? e - 1 : e);
        break;
      }
      case s.UP: {
        l.preventDefault(), n((e) => e > 0 ? e - 1 : e);
        break;
      }
      case s.DOWN: {
        l.preventDefault(), n((e) => e < a - 1 ? e + 1 : e);
        break;
      }
      case s.HOME: {
        l.preventDefault(), l.ctrlKey && n(0), i(0);
        break;
      }
      case s.END: {
        l.preventDefault(), l.ctrlKey && n(a - 1), i(t - 1);
        break;
      }
      case s.PAGE_DOWN: {
        l.preventDefault(), n((e) => e + r < a ? e + r : a - 1);
        break;
      }
      case s.PAGE_UP: {
        l.preventDefault(), n((e) => e - r > 0 ? e - r : 0);
        break;
      }
    }
  };
  return /* @__PURE__ */ K.createElement(F.Provider, {
    value: { rowIndex: D, colIndex: k, setTableValues: O }
  }, /* @__PURE__ */ K.createElement("table", P({
    role: "grid",
    ref: m,
    "aria-rowcount": a,
    "aria-colcount": t,
    onKeyDown: h
  }, N)));
};
_.defaultProps = {
  jumpStep: 3,
  initialCol: 0,
  initialRow: 0
};
_.propTypes = {
  colCount: u.number.isRequired,
  initialCol: u.number,
  initialRow: u.number,
  jumpStep: u.number,
  rowCount: u.number.isRequired
};
export {
  _ as RawTable
};
