// @strapi/design-system/dist/FocusTrap/ExampleComponent.js
import e, { useState as l } from "react";
import m from "@strapi/icons/Cross";
import { Box as t } from "../Box/Box.js";
import { Stack as s } from "../Stack/Stack.js";
import { Typography as i } from "../Typography/Typography.js";
import { Button as r } from "../Button/Button.js";
import { Flex as o } from "../Flex/Flex.js";
import { FocusTrap as u } from "./FocusTrap.js";
const d = ({ onClose: a }) => {
  const [n, c] = l(!1);
  return /* @__PURE__ */ e.createElement(u, {
    onEscape: a
  }, /* @__PURE__ */ e.createElement(t, {
    background: "neutral0",
    padding: 4,
    hasRadius: !0,
    style: { width: "600px" }
  }, /* @__PURE__ */ e.createElement(s, {
    spacing: 2
  }, /* @__PURE__ */ e.createElement(o, {
    justifyContent: "space-between"
  }, /* @__PURE__ */ e.createElement(i, {
    variant: "beta",
    as: "h2"
  }, "Hey folks!"), /* @__PURE__ */ e.createElement("button", {
    type: "button",
    style: { border: "none", background: "transparent" },
    onClick: a,
    "aria-label": "Close"
  }, /* @__PURE__ */ e.createElement(m, {
    "aria-hidden": !0
  }))), /* @__PURE__ */ e.createElement(t, {
    paddingTop: 2,
    paddingBottom: 2
  }, /* @__PURE__ */ e.createElement(i, null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.")), /* @__PURE__ */ e.createElement(o, {
    justifyContent: "space-between"
  }, /* @__PURE__ */ e.createElement(r, {
    id: "second"
  }, "Second focusable"), /* @__PURE__ */ e.createElement(r, {
    id: "last",
    onClick: () => c(!0)
  }, "Last focusable (at the beginning)"), n && /* @__PURE__ */ e.createElement(r, {
    id: "real-last"
  }, "Last focusable (at the end)")))));
}, y = () => {
  const [a, n] = l(!1);
  return /* @__PURE__ */ e.createElement(t, {
    background: "neutral150",
    padding: 10
  }, /* @__PURE__ */ e.createElement(s, {
    spacing: 2
  }, a && /* @__PURE__ */ e.createElement(d, {
    onClose: () => n(!1)
  }), /* @__PURE__ */ e.createElement(t, {
    background: "neutral0",
    padding: 4,
    hasRadius: !0,
    style: { width: "600px" }
  }, /* @__PURE__ */ e.createElement(i, {
    variant: "beta",
    as: "h2"
  }, "Outside the trap!"), /* @__PURE__ */ e.createElement(t, {
    paddingTop: 2
  }, /* @__PURE__ */ e.createElement(r, {
    onClick: () => n(!0),
    id: "trigger"
  }, "Open the trap")))));
};
export {
  y as ExampleComponent
};
