// @strapi/design-system/dist/v2/Pagination/components.js
var R = Object.defineProperty, T = Object.defineProperties;
var N = Object.getOwnPropertyDescriptors;
var m = Object.getOwnPropertySymbols;
var y = Object.prototype.hasOwnProperty, P = Object.prototype.propertyIsEnumerable;
var b = (e, r, o) => r in e ? R(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, l = (e, r) => {
  for (var o in r || (r = {}))
    y.call(r, o) && b(e, o, r[o]);
  if (m)
    for (var o of m(r))
      P.call(r, o) && b(e, o, r[o]);
  return e;
}, f = (e, r) => T(e, N(r));
var c = (e, r) => {
  var o = {};
  for (var t in e)
    y.call(e, t) && r.indexOf(t) < 0 && (o[t] = e[t]);
  if (e != null && m)
    for (var t of m(e))
      r.indexOf(t) < 0 && P.call(e, t) && (o[t] = e[t]);
  return o;
};
import a from "react";
import p from "styled-components";
import u from "prop-types";
import W from "@strapi/icons/ChevronLeft";
import q from "@strapi/icons/ChevronRight";
import { VisuallyHidden as h } from "../../VisuallyHidden/VisuallyHidden.js";
import { usePagination as v } from "./PaginationContext.js";
import { Typography as C } from "../../Typography/Typography.js";
import { buttonFocusStyle as S } from "../../themes/utils.js";
import { BaseLink as A } from "../../BaseLink/BaseLink.js";
const E = p(C)`
  line-height: revert;
`, D = {
  active: !0
}, g = p(A).withConfig({
  shouldForwardProp: (e, r) => !D[e] && r(e)
})`
  padding: ${({ theme: e }) => e.spaces[3]};
  border-radius: ${({ theme: e }) => e.borderRadius};
  box-shadow: ${({ active: e, theme: r }) => e ? r.shadows.filterShadow : void 0};
  text-decoration: none;
  display: flex;

  ${S}
`, F = p(g)`
  color: ${({ theme: e, active: r }) => r ? e.colors.primary700 : e.colors.neutral800};
  background: ${({ theme: e, active: r }) => r ? e.colors.neutral0 : void 0};

  &:hover {
    box-shadow: ${({ theme: e }) => e.shadows.filterShadow};
  }
`, k = p(g)`
  font-size: 0.7rem;
  svg path {
    fill: ${(e) => e["aria-disabled"] ? e.theme.colors.neutral300 : e.theme.colors.neutral600};
  }

  &:focus,
  &:hover {
    svg path {
      fill: ${(e) => e["aria-disabled"] ? e.theme.colors.neutral300 : e.theme.colors.neutral700};
    }
  }

  ${(e) => e["aria-disabled"] ? `
  pointer-events: none;
    ` : void 0}
`, I = p(g)`
  color: ${({ theme: e }) => e.colors.neutral800};
`, w = a.forwardRef((t, o) => {
  var i = t, { children: e } = i, r = c(i, ["children"]);
  const { activePage: n } = v(), s = n === 1;
  return /* @__PURE__ */ a.createElement(k, l({
    ref: o,
    "aria-disabled": s,
    tabIndex: s ? -1 : void 0
  }, r), /* @__PURE__ */ a.createElement(h, null, e), /* @__PURE__ */ a.createElement(W, {
    "aria-hidden": !0
  }));
});
w.displayName = "PreviousLink";
const L = a.forwardRef((t, o) => {
  var i = t, { children: e } = i, r = c(i, ["children"]);
  const { activePage: n, pageCount: s } = v(), d = n === s;
  return /* @__PURE__ */ a.createElement(k, l({
    ref: o,
    "aria-disabled": d,
    tabIndex: d ? -1 : void 0
  }, r), /* @__PURE__ */ a.createElement(h, null, e), /* @__PURE__ */ a.createElement(q, {
    "aria-hidden": !0
  }));
});
L.displayName = "NextLink";
const $ = a.forwardRef((i, t) => {
  var n = i, { number: e, children: r } = n, o = c(n, ["number", "children"]);
  const { activePage: s } = v(), d = s === e;
  return /* @__PURE__ */ a.createElement(F, f(l({
    ref: t
  }, o), {
    active: d
  }), /* @__PURE__ */ a.createElement(h, null, r), /* @__PURE__ */ a.createElement(E, {
    "aria-hidden": !0,
    variant: "pi",
    fontWeight: d ? "bold" : null
  }, e));
});
$.displayName = "PageLink";
const _ = (o) => {
  var t = o, { children: e } = t, r = c(t, ["children"]);
  return /* @__PURE__ */ a.createElement(I, f(l({}, r), {
    as: "div"
  }), /* @__PURE__ */ a.createElement(h, null, e), /* @__PURE__ */ a.createElement(E, {
    "aria-hidden": !0,
    variant: "pi"
  }, "\u2026"));
};
$.propTypes = {
  children: u.node.isRequired,
  number: u.number.isRequired
};
const x = {
  children: u.node.isRequired
};
L.propTypes = x;
w.propTypes = x;
_.propTypes = {
  children: u.node.isRequired
};
export {
  _ as Dots,
  L as NextLink,
  $ as PageLink,
  w as PreviousLink
};
