// @strapi/design-system/dist/Switch/Switch.cjs
"use strict";var x=Object.defineProperty;var l=Object.getOwnPropertySymbols;var b=Object.prototype.hasOwnProperty,m=Object.prototype.propertyIsEnumerable;var p=(e,r,t)=>r in e?x(e,r,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[r]=t,g=(e,r)=>{for(var t in r||(r={}))b.call(r,t)&&p(e,t,r[t]);if(l)for(var t of l(r))m.call(r,t)&&p(e,t,r[t]);return e};var h=(e,r)=>{var t={};for(var a in e)b.call(e,a)&&r.indexOf(a)<0&&(t[a]=e[a]);if(e!=null&&l)for(var a of l(e))r.indexOf(a)<0&&m.call(e,a)&&(t[a]=e[a]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const _=require("react"),k=require("prop-types"),q=require("styled-components"),S=require("../Flex/Flex.cjs"),E=require("../Box/Box.cjs"),c=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},o=c(_),n=c(k),y=c(q),d=y.default.div`
  background: ${({theme:e})=>e.colors.danger500};
  border: none;
  border-radius: 16px;
  position: relative;
  height: ${24/16}rem;
  width: ${40/16}rem;

  & span {
    font-size: ${({visibleLabels:e})=>e?"1rem":0};
  }

  &:before {
    content: '';
    background: ${({theme:e})=>e.colors.neutral0};
    width: 1rem;
    height: 1rem;
    border-radius: 50%;
    position: absolute;
    transition: all 0.5s;
    left: ${({theme:e})=>e.spaces[1]};
    top: ${({theme:e})=>e.spaces[1]};
  }

  @media (prefers-reduced-motion: reduce) {
    &:before {
      transition: none;
    }
  }
`,R=y.default.button`
  background: transparent;
  padding: 0;
  border: none;

  &[aria-checked='true'] ${d} {
    background: ${({theme:e})=>e.colors.success500};
  }

  &[aria-checked='true'] ${d}:before {
    transform: translateX(1rem);
  }
`,s=o.default.forwardRef((v,w)=>{var f=v,{label:e,onChange:r,onLabel:t,offLabel:a,selected:u,visibleLabels:i}=f,$=h(f,["label","onChange","onLabel","offLabel","selected","visibleLabels"]);return o.default.createElement(R,g({ref:w,role:"switch","aria-checked":u,"aria-label":e,onClick:r,visibleLabels:i,type:"button"},$),o.default.createElement(S.Flex,null,o.default.createElement(d,null,o.default.createElement("span",null,t),o.default.createElement("span",null,a)),i&&o.default.createElement(E.Box,{as:"span","aria-hidden":!0,paddingLeft:2,color:u?"success600":"danger600"},u?t:a)))});s.defaultProps={onLabel:"On",offLabel:"Off",visibleLabels:!1};s.propTypes={label:n.default.string.isRequired,offLabel:n.default.string,onChange:n.default.func.isRequired,onLabel:n.default.string,selected:n.default.bool.isRequired,visibleLabels:n.default.bool};s.displayName="Switch";exports.Switch=s;
