// @strapi/design-system/dist/CarouselInput/CarouselImage.js
var d = Object.defineProperty, g = Object.defineProperties;
var u = Object.getOwnPropertyDescriptors;
var m = Object.getOwnPropertySymbols;
var E = Object.prototype.hasOwnProperty, y = Object.prototype.propertyIsEnumerable;
var l = (e, r, t) => r in e ? d(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, o = (e, r) => {
  for (var t in r || (r = {}))
    E.call(r, t) && l(e, t, r[t]);
  if (m)
    for (var t of m(r))
      y.call(r, t) && l(e, t, r[t]);
  return e;
}, a = (e, r) => g(e, u(r));
import s, { useState as I } from "react";
import n from "prop-types";
import h from "styled-components";
import { Box as S } from "../Box/Box.js";
import { Tooltip as T } from "../Tooltip/Tooltip.js";
import { ellipsisStyle as v } from "../Typography/utils.js";
const p = h(S)`
  height: 100%;
  ${v({ ellipsis: !0 })}
`, c = (e) => {
  var i;
  const [r, t] = I(!1), f = () => {
    t(!0);
  };
  return r ? /* @__PURE__ */ s.createElement(T, {
    description: (i = e.alt) != null ? i : ""
  }, /* @__PURE__ */ s.createElement(p, o({
    as: "img"
  }, e))) : /* @__PURE__ */ s.createElement(p, a(o({
    as: "img"
  }, e), {
    onError: f
  }));
};
c.defaultProps = {
  src: void 0,
  alt: void 0
};
c.propTypes = {
  alt: n.string,
  src: n.string
};
export {
  c as CarouselImage
};
