// @strapi/design-system/dist/Loader/Loader.js
var g = Object.defineProperty;
var t = Object.getOwnPropertySymbols;
var n = Object.prototype.hasOwnProperty, d = Object.prototype.propertyIsEnumerable;
var m = (e, r, a) => r in e ? g(e, r, { enumerable: !0, configurable: !0, writable: !0, value: a }) : e[r] = a, f = (e, r) => {
  for (var a in r || (r = {}))
    n.call(r, a) && m(e, a, r[a]);
  if (t)
    for (var a of t(r))
      d.call(r, a) && m(e, a, r[a]);
  return e;
};
var p = (e, r) => {
  var a = {};
  for (var o in e)
    n.call(e, o) && r.indexOf(o) < 0 && (a[o] = e[o]);
  if (e != null && t)
    for (var o of t(e))
      r.indexOf(o) < 0 && d.call(e, o) && (a[o] = e[o]);
  return a;
};
import i, { forwardRef as u } from "react";
import c from "prop-types";
import y, { keyframes as h } from "styled-components";
import { VisuallyHidden as v } from "../VisuallyHidden/VisuallyHidden.js";
import $ from "./assets/loader.js";
const w = h`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`, E = y.img`
  animation: ${w} 1s infinite linear;
  will-change: transform;
  ${({ small: e, theme: r }) => e && `width: ${r.spaces[6]}; height: ${r.spaces[6]};`}
`, l = u((L, o) => {
  var s = L, { children: e, small: r } = s, a = p(s, ["children", "small"]);
  return /* @__PURE__ */ i.createElement("div", f({
    role: "alert",
    "aria-live": "assertive",
    ref: o
  }, a), /* @__PURE__ */ i.createElement(v, null, e), /* @__PURE__ */ i.createElement(E, {
    src: $,
    "aria-hidden": !0,
    small: r
  }));
});
l.displayName = "Loader";
l.defaultProps = {
  small: !1
};
l.propTypes = {
  children: c.node.isRequired,
  small: c.bool
};
export {
  l as Loader
};
