// @strapi/design-system/dist/Dialog/Dialog.js
var D = Object.defineProperty;
var l = Object.getOwnPropertySymbols;
var u = Object.prototype.hasOwnProperty, c = Object.prototype.propertyIsEnumerable;
var p = (e, r, o) => r in e ? D(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, f = (e, r) => {
  for (var o in r || (r = {}))
    u.call(r, o) && p(e, o, r[o]);
  if (l)
    for (var o of l(r))
      c.call(r, o) && p(e, o, r[o]);
  return e;
};
var g = (e, r) => {
  var o = {};
  for (var t in e)
    u.call(e, t) && r.indexOf(t) < 0 && (o[t] = e[t]);
  if (e != null && l)
    for (var t of l(e))
      r.indexOf(t) < 0 && c.call(e, t) && (o[t] = e[t]);
  return o;
};
import i from "prop-types";
import a from "react";
import n from "styled-components";
import { Box as y } from "../Box/Box.js";
import { FocusTrap as x } from "../FocusTrap/FocusTrap.js";
import { Portal as w } from "../Portal/Portal.js";
import { Flex as R } from "../Flex/Flex.js";
import { Typography as T } from "../Typography/Typography.js";
import { DismissibleLayer as $ } from "../DismissibleLayer/DismissibleLayer.js";
import { setOpacity as v } from "../helpers/setOpacity.js";
import P from "../helpers/useLockScroll.js";
import { useId as k } from "../helpers/useId.js";
const q = n.div`
  position: fixed;
  z-index: 4;
  inset: 0;
  background: ${({ theme: e }) => v(e.colors.neutral800, 0.2)};
  padding: 0 ${({ theme: e }) => e.spaces[8]};
`, O = n(y)`
  max-width: ${412 / 16}rem;
  margin: 0 auto;
  overflow: hidden;
  margin-top: 10%;
`, B = n(R)`
  border-bottom: 1px solid ${({ theme: e }) => e.colors.neutral150};
`, m = (C) => {
  var s = C, { onClose: e, title: r, as: o, isOpen: t, id: b } = s, E = g(s, ["onClose", "title", "as", "isOpen", "id"]);
  const h = k("dialog", b);
  if (P(t), !t)
    return null;
  const d = `${h}-label`;
  return /* @__PURE__ */ a.createElement(w, null, /* @__PURE__ */ a.createElement(q, null, /* @__PURE__ */ a.createElement(x, null, /* @__PURE__ */ a.createElement($, {
    onEscapeKeyDown: e,
    onPointerDownOutside: e
  }, /* @__PURE__ */ a.createElement(O, {
    "aria-labelledby": d,
    "aria-modal": !0,
    background: "neutral0",
    hasRadius: !0,
    shadow: "popupShadow",
    role: "dialog"
  }, /* @__PURE__ */ a.createElement(B, {
    padding: 6,
    justifyContent: "center"
  }, /* @__PURE__ */ a.createElement(T, {
    variant: "beta",
    as: o || "h2",
    id: d
  }, r)), /* @__PURE__ */ a.createElement(y, f({}, E)))))));
};
m.displayName = "Dialog";
m.defaultProps = {
  as: "h2",
  id: void 0
};
m.propTypes = {
  as: i.string,
  id: i.oneOfType([i.string, i.number]),
  isOpen: i.bool.isRequired,
  onClose: i.func.isRequired,
  title: i.string.isRequired
};
export {
  m as Dialog
};
