// @strapi/design-system/dist/Select/Select.js
var se = Object.defineProperty;
var B = Object.getOwnPropertySymbols;
var G = Object.prototype.hasOwnProperty, H = Object.prototype.propertyIsEnumerable;
var z = (o, l, i) => l in o ? se(o, l, { enumerable: !0, configurable: !0, writable: !0, value: i }) : o[l] = i, W = (o, l) => {
  for (var i in l || (l = {}))
    G.call(l, i) && z(o, i, l[i]);
  if (B)
    for (var i of B(l))
      H.call(l, i) && z(o, i, l[i]);
  return o;
};
var N = (o, l) => {
  var i = {};
  for (var s in o)
    G.call(o, s) && l.indexOf(s) < 0 && (i[s] = o[s]);
  if (o != null && B)
    for (var s of B(o))
      l.indexOf(s) < 0 && H.call(o, s) && (i[s] = o[s]);
  return i;
};
import r, { useState as ae, useRef as pe, Children as J, cloneElement as K } from "react";
import e from "prop-types";
import ce from "@strapi/icons/CarretDown";
import me from "@strapi/icons/Cross";
import de from "styled-components";
import { sizes as fe } from "../themes/sizes.js";
import { SelectButton as ue } from "./SelectButton.js";
import { Field as he } from "../Field/Field.js";
import { FieldLabel as Ee } from "../Field/FieldLabel.js";
import "../Field/FieldInput.js";
import { FieldHint as be } from "../Field/FieldHint.js";
import { FieldError as ye } from "../Field/FieldError.js";
import "../Field/FieldContext.js";
import "../Field/FieldAction.js";
import { Popover as ge } from "../Popover/Popover.js";
import { Stack as ve } from "../Stack/Stack.js";
import { Typography as Q } from "../Typography/Typography.js";
import { Flex as $ } from "../Flex/Flex.js";
import { Box as U } from "../Box/Box.js";
import { useId as Be } from "../helpers/useId.js";
import { SelectList as Oe } from "./SelectList.js";
import { SelectButtonWrapper as Te, IconBox as Ce, CaretBox as Ie } from "./components.js";
import { useButtonRef as Se } from "./hooks/useButtonRef.js";
import { VisuallyHidden as X } from "../VisuallyHidden/VisuallyHidden.js";
import { DownState as xe } from "./constants.js";
import { escapeSelector as Y } from "../helpers/escapeSelector.js";
import { SelectTags as $e } from "./SelectTags.js";
const Re = de($)`
  width: 100%;
`, Z = (we) => {
  var L = we, {
    label: o,
    labelAction: l,
    id: i,
    children: s,
    customizeContent: h,
    placeholder: O,
    onChange: T,
    value: n,
    hint: C,
    error: E,
    disabled: m,
    clearLabel: R,
    onClear: w,
    onReachEnd: V,
    multi: p,
    required: ee,
    selectButtonTitle: te,
    size: re,
    startIcon: D,
    withTags: b
  } = L, oe = N(L, [
    "label",
    "labelAction",
    "id",
    "children",
    "customizeContent",
    "placeholder",
    "onChange",
    "value",
    "hint",
    "error",
    "disabled",
    "clearLabel",
    "onClear",
    "onReachEnd",
    "multi",
    "required",
    "selectButtonTitle",
    "size",
    "startIcon",
    "withTags"
  ]);
  const a = Be("select", i), [y, g] = ae(void 0), F = Se(y), M = pe(null), k = `${a}-label`, v = `${a}-content`, j = typeof E == "string";
  let I;
  if (j ? I = `${a}-error` : C && (I = `${a}-hint`), b && !p)
    throw new Error('The "withTags" props can only be used when the "multi" prop is present');
  const ne = () => {
    g(void 0);
  }, ie = () => {
    m || (w(), F.current.focus());
  }, A = (t) => {
    t.preventDefault(), !m && (t.nativeEvent.which === 3 || t.nativeEvent.button === 2 || g(xe.Mouse));
  }, S = (t, c = !0) => {
    p ? T(n.includes(t) ? n.filter((d) => d !== t) : [...n, t]) : (T(t), c && g(void 0));
  }, P = (t) => {
    T(
      n.includes(t[0]) ? n.filter((c, d, u) => u.indexOf(c) < 0, t) : [...n, ...t]
    );
  };
  let x, _ = [];
  const q = (t, c) => {
    const d = `${a}-option-${t.props.value}`, u = p ? n.includes(t.props.value) : t.props.value === n;
    return u && (b ? _.push({ label: t.props.children, value: t.props.value }) : x = t.props.children), K(t, {
      id: Y(d),
      onClick: () => S(t.props.value),
      selected: u,
      multi: p,
      isChild: c
    });
  }, le = J.toArray(s).map((t) => {
    if (t.type.displayName === "OptGroup") {
      const c = `${a}-option-${t.props.label}`, d = t.props.children.every((f) => n.includes(f.props.value)), u = !d && t.props.children.some((f) => n.includes(f.props.value));
      return K(t, {
        id: Y(c),
        onClick: () => P(t.props.children.map((f) => f.props.value)),
        selected: d,
        indeterminate: u,
        multi: p,
        children: J.toArray(t.props.children).map((f) => q(f, !0)),
        value: t.props.label
      });
    }
    return q(t);
  });
  return /* @__PURE__ */ r.createElement(he, {
    hint: C,
    error: E,
    id: a,
    required: ee
  }, /* @__PURE__ */ r.createElement(ve, {
    spacing: o || C || j ? 1 : 0
  }, o && /* @__PURE__ */ r.createElement(Ee, {
    as: "span",
    action: l
  }, o), /* @__PURE__ */ r.createElement(Te, {
    size: re,
    hasError: Boolean(E),
    disabled: m,
    ref: M
  }, /* @__PURE__ */ r.createElement(ue, W({
    ref: F,
    labelledBy: `${k} ${v}`,
    "aria-describedby": I,
    expanded: Boolean(y),
    onTrigger: g,
    id: a,
    hasError: Boolean(E),
    disabled: m,
    onMouseDown: A
  }, oe)), /* @__PURE__ */ r.createElement(Re, {
    justifyContent: "space-between"
  }, /* @__PURE__ */ r.createElement($, null, D && /* @__PURE__ */ r.createElement(U, {
    paddingLeft: 3,
    "aria-hidden": !0
  }, D), b && /* @__PURE__ */ r.createElement($e, {
    tags: _,
    onRemoveTag: S,
    disabled: m
  }), /* @__PURE__ */ r.createElement(U, {
    paddingLeft: 4,
    paddingRight: 4
  }, b ? /* @__PURE__ */ r.createElement(r.Fragment, null, !n || n.length === 0 ? /* @__PURE__ */ r.createElement(Q, {
    ellipsis: !0,
    id: v,
    textColor: "neutral600"
  }, O) : null, /* @__PURE__ */ r.createElement(X, {
    as: "span",
    id: v
  }, h ? h(n) : x || O, n.join(", "))) : /* @__PURE__ */ r.createElement(Q, {
    ellipsis: !0,
    id: v,
    textColor: n ? "neutral800" : "neutral600"
  }, h ? h(n) : x || O, p && /* @__PURE__ */ r.createElement(X, {
    as: "span"
  }, n.join(", "))))), /* @__PURE__ */ r.createElement($, null, (p && n && n.length || !p && n) && w && /* @__PURE__ */ r.createElement(Ce, {
    as: "button",
    type: "button",
    onClick: ie,
    "aria-label": R,
    "aria-disabled": m,
    title: R
  }, /* @__PURE__ */ r.createElement(me, null)), /* @__PURE__ */ r.createElement(Ie, {
    paddingLeft: 3,
    "aria-hidden": !0,
    as: "button",
    type: "button",
    onMouseDown: A,
    tabIndex: -1,
    disabled: m,
    title: te
  }, /* @__PURE__ */ r.createElement(ce, null))))), /* @__PURE__ */ r.createElement(be, null), /* @__PURE__ */ r.createElement(ye, null)), y && /* @__PURE__ */ r.createElement(ge, {
    source: M,
    spacing: 4,
    fullWidth: !0,
    intersectionId: `select-list-intersection-${a}`,
    onReachEnd: V
  }, /* @__PURE__ */ r.createElement(Oe, {
    selectId: a,
    labelledBy: o ? k : void 0,
    onEscape: ne,
    expanded: y,
    onSelectItem: (t, c) => c ? P(t) : S(t, !1),
    multi: p
  }, le)));
};
Z.defaultProps = {
  "aria-label": void 0,
  children: [],
  clearLabel: "Clear",
  customizeContent: void 0,
  disabled: !1,
  id: void 0,
  label: void 0,
  labelAction: void 0,
  multi: !1,
  onChange() {
  },
  onClear: void 0,
  onReachEnd: void 0,
  value: void 0,
  hint: void 0,
  error: void 0,
  placeholder: "Select...",
  required: !1,
  selectButtonTitle: "Carret Down Button",
  size: "M",
  startIcon: void 0,
  withTags: !1
};
Z.propTypes = {
  "aria-label": e.string,
  children: e.oneOfType([e.arrayOf(e.node), e.node]),
  clearLabel: e.string,
  customizeContent: e.func,
  disabled: e.bool,
  error: e.oneOfType([e.string, e.bool]),
  hint: e.oneOfType([e.string, e.node, e.arrayOf(e.node)]),
  id: e.oneOfType([e.string, e.number]),
  label: e.string,
  labelAction: e.element,
  multi: e.bool,
  onChange: e.func,
  onClear: e.func,
  onReachEnd: e.func,
  placeholder: e.string,
  required: e.bool,
  selectButtonTitle: e.string,
  size: e.oneOf(Object.keys(fe.input)),
  startIcon: e.element,
  value: e.oneOfType([
    e.arrayOf(e.oneOfType([e.string, e.number])),
    e.string,
    e.number
  ]),
  withTags: e.bool
};
export {
  Z as Select
};
