// @strapi/design-system/dist/Flex/FlexProps.js
var a = Object.defineProperty;
var t = Object.getOwnPropertySymbols;
var l = Object.prototype.hasOwnProperty, m = Object.prototype.propertyIsEnumerable;
var i = (o, r, n) => r in o ? a(o, r, { enumerable: !0, configurable: !0, writable: !0, value: n }) : o[r] = n, s = (o, r) => {
  for (var n in r || (r = {}))
    l.call(r, n) && i(o, n, r[n]);
  if (t)
    for (var n of t(r))
      m.call(r, n) && i(o, n, r[n]);
  return o;
};
import f from "react";
import e from "prop-types";
const p = (o) => /* @__PURE__ */ f.createElement("div", s({}, o)), b = {
  alignItems: "center",
  basis: void 0,
  direction: "row",
  gap: void 0,
  inline: !1,
  justifyContent: void 0,
  reverse: !1,
  shrink: void 0,
  wrap: void 0
}, u = {
  alignItems: e.string,
  basis: e.oneOfType([e.string, e.number]),
  direction: e.string,
  gap: e.oneOfType([
    e.shape({
      desktop: e.number,
      mobile: e.number,
      tablet: e.number
    }),
    e.number,
    e.arrayOf(e.number),
    e.string
  ]),
  inline: e.bool,
  justifyContent: e.string,
  reverse: e.bool,
  shrink: e.number,
  wrap: e.string
};
p.defaultProps = b;
p.propTypes = u;
export {
  p as FlexProps,
  b as flexDefaultProps,
  u as flexPropTypes
};
