// @strapi/design-system/dist/Field/FieldLabel.js
var b = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var s = Object.prototype.hasOwnProperty, p = Object.prototype.propertyIsEnumerable;
var d = (e, o, r) => o in e ? b(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, m = (e, o) => {
  for (var r in o || (o = {}))
    s.call(o, r) && d(e, r, o[r]);
  if (i)
    for (var r of i(o))
      p.call(o, r) && d(e, r, o[r]);
  return e;
};
var c = (e, o) => {
  var r = {};
  for (var t in e)
    s.call(e, t) && o.indexOf(t) < 0 && (r[t] = e[t]);
  if (e != null && i)
    for (var t of i(e))
      o.indexOf(t) < 0 && p.call(e, t) && (r[t] = e[t]);
  return r;
};
import n from "react";
import u from "styled-components";
import l from "prop-types";
import { Flex as h } from "../Flex/Flex.js";
import { useField as v } from "./FieldContext.js";
import { Typography as f } from "../Typography/Typography.js";
const x = u(f)`
  line-height: 0;
`, T = u(h)`
  line-height: 0;

  svg path {
    fill: ${({ theme: e }) => e.colors.neutral500};
  }
`, g = (E) => {
  var a = E, { children: e, action: o, required: r } = a, t = c(a, ["children", "action", "required"]);
  const { id: q, required: y } = v(), F = y || r;
  return r !== void 0 && console.warn(
    'Deprecation warning: Usage of "required" prop in FieldLabel component is deprecated. This is discouraged and will be removed in the next major release. Please use the Field component to share the required prop.'
  ), /* @__PURE__ */ n.createElement(f, m({
    variant: "pi",
    textColor: "neutral800",
    htmlFor: q,
    fontWeight: "bold",
    as: "label"
  }, t), /* @__PURE__ */ n.createElement(h, {
    alignItems: "center"
  }, e, F && /* @__PURE__ */ n.createElement(x, {
    textColor: "danger600"
  }, "*"), o && /* @__PURE__ */ n.createElement(T, {
    marginLeft: 1
  }, o)));
};
g.defaultProps = {
  action: void 0,
  required: void 0
};
g.propTypes = {
  action: l.element,
  children: l.node.isRequired,
  required: l.bool
};
export {
  g as FieldLabel
};
