// @strapi/design-system/dist/Field/FieldInput.cjs
"use strict";var z=Object.defineProperty;var n=Object.getOwnPropertySymbols;var h=Object.prototype.hasOwnProperty,$=Object.prototype.propertyIsEnumerable;var b=(e,t,o)=>t in e?z(e,t,{enumerable:!0,configurable:!0,writable:!0,value:o}):e[t]=o,y=(e,t)=>{for(var o in t||(t={}))h.call(t,o)&&b(e,o,t[o]);if(n)for(var o of n(t))$.call(t,o)&&b(e,o,t[o]);return e};var m=(e,t)=>{var o={};for(var r in e)h.call(e,r)&&t.indexOf(r)<0&&(o[r]=e[r]);if(e!=null&&n)for(var r of n(e))t.indexOf(r)<0&&$.call(e,r)&&(o[r]=e[r]);return o};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const q=require("react"),D=require("styled-components"),E=require("prop-types"),M=require("../themes/sizes.cjs"),O=require("../themes/utils.cjs"),P=require("./FieldContext.cjs"),T=require("../Flex/Flex.cjs"),x=require("../Box/Box.cjs"),u=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},s=u(q),_=u(D),a=u(E),F={S:6.5,M:10.5},L=_.default.input`
  border: none;
  border-radius: ${({theme:e})=>e.borderRadius};
  padding-bottom: ${({size:e})=>`${F[e]/16}rem`};
  padding-left: ${({theme:e,hasLeftAction:t})=>t?0:e.spaces[4]};
  padding-right: ${({theme:e,hasRightAction:t})=>t?0:e.spaces[4]};
  padding-top: ${({size:e})=>`${F[e]/16}rem`};
  cursor: ${e=>e["aria-disabled"]?"not-allowed":void 0};

  color: ${({theme:e})=>e.colors.neutral800};
  font-weight: 400;
  // TODO: Make sure to use the theme when it's ready
  font-size: ${14/16}rem;
  display: block;
  width: 100%;
  background: inherit;

  ::placeholder {
    color: ${({theme:e})=>e.colors.neutral500};
    opacity: 1;
  }

  &[aria-disabled='true'] {
    color: inherit;
  }

  //focus managed by InputWrapper
  &:focus {
    outline: none;
    box-shadow: none;
  }
`,j=_.default(T.Flex)`
  border: 1px solid ${({theme:e,hasError:t})=>t?e.colors.danger600:e.colors.neutral200};
  border-radius: ${({theme:e})=>e.borderRadius};
  background: ${({theme:e})=>e.colors.neutral0};
  ${O.inputFocusStyle()}

  ${({theme:e,disabled:t})=>t?`
    color: ${e.colors.neutral600};
    background: ${e.colors.neutral150};
  
  `:void 0}
`,i=q.forwardRef((S,I)=>{var p=S,{endAction:e,startAction:t,disabled:o,onChange:r,size:c}=p,B=m(p,["endAction","startAction","disabled","onChange","size"]);const{id:d,error:f,hint:R,name:v,required:w}=P.useField();let l;f?l=`${d}-error`:R&&(l=`${d}-hint`);const g=Boolean(f),C=k=>{o||r(k)};return s.default.createElement(j,{size:c,justifyContent:"space-between",hasError:g,disabled:o},t&&s.default.createElement(x.Box,{paddingLeft:3,paddingRight:2},t),s.default.createElement(L,y({id:d,name:v,ref:I,"aria-describedby":l,"aria-invalid":g,"aria-disabled":o,hasLeftAction:Boolean(t),hasRightAction:Boolean(e),onChange:C,"aria-required":w,size:c},B)),e&&s.default.createElement(x.Box,{paddingLeft:2,paddingRight:3},e))});i.displayName="FieldInput";i.defaultProps={disabled:!1,endAction:void 0,size:"M",startAction:void 0,onChange(){}};i.propTypes={disabled:a.default.bool,endAction:a.default.element,onChange:a.default.func,size:a.default.oneOf(Object.keys(M.sizes.input)),startAction:a.default.element};exports.FieldInput=i;exports.InputWrapper=j;
