// @strapi/design-system/dist/Field/Field.js
var T = Object.defineProperty;
var t = Object.getOwnPropertySymbols;
var m = Object.prototype.hasOwnProperty, a = Object.prototype.propertyIsEnumerable;
var p = (e, o, r) => o in e ? T(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, f = (e, o) => {
  for (var r in o || (o = {}))
    m.call(o, r) && p(e, r, o[r]);
  if (t)
    for (var r of t(o))
      a.call(o, r) && p(e, r, o[r]);
  return e;
};
var l = (e, o) => {
  var r = {};
  for (var d in e)
    m.call(e, d) && o.indexOf(d) < 0 && (r[d] = e[d]);
  if (e != null && t)
    for (var d of t(e))
      o.indexOf(d) < 0 && a.call(e, d) && (r[d] = e[d]);
  return r;
};
import s from "react";
import i from "prop-types";
import { FieldContext as O } from "./FieldContext.js";
import { useId as b } from "../helpers/useId.js";
const u = (h) => {
  var n = h, { children: e, name: o, error: r, hint: d, id: v, required: c } = n, y = l(n, ["children", "name", "error", "hint", "id", "required"]);
  const g = b("field", v);
  return /* @__PURE__ */ s.createElement("div", f({}, y), /* @__PURE__ */ s.createElement(O.Provider, {
    value: { name: o, id: g, error: r, hint: d, required: c }
  }, e));
};
u.defaultProps = {
  error: void 0,
  hint: void 0,
  id: void 0,
  name: void 0,
  required: !1
};
u.propTypes = {
  children: i.node.isRequired,
  error: i.oneOfType([i.string, i.bool]),
  hint: i.oneOfType([i.string, i.node, i.arrayOf(i.node)]),
  id: i.oneOfType([i.string, i.number]),
  name: i.string,
  required: i.bool
};
export {
  u as Field
};
