// @strapi/design-system/dist/Accordion/AccordionGroup.js
import o, { Children as b, cloneElement as m } from "react";
import t from "prop-types";
import a from "styled-components";
import { Box as d } from "../Box/Box.js";
import { Typography as n } from "../Typography/Typography.js";
import { Flex as $ } from "../Flex/Flex.js";
import { KeyboardNavigable as f } from "../KeyboardNavigable/KeyboardNavigable.js";
const g = a(d)`
  border-bottom: 1px solid ${({ theme: r }) => r.colors.neutral200};
  border-right: 1px solid ${({ theme: r }) => r.colors.neutral200};
  border-left: 1px solid ${({ theme: r }) => r.colors.neutral200};
  border-radius: 0 0 ${({ theme: r }) => r.borderRadius} ${({ theme: r }) => r.borderRadius};
`, x = a(d)`
  & > * {
    & > * {
      border-radius: unset;
    }
  }

  & > * {
    border-radius: unset;
    border-right: 1px solid ${({ theme: r }) => r.colors.neutral200};
    border-left: 1px solid ${({ theme: r }) => r.colors.neutral200};
    border-bottom: 1px solid ${({ theme: r }) => r.colors.neutral200};
  }

  & > *:first-of-type {
    border-top: 1px solid ${({ theme: r }) => r.colors.neutral200};
    border-radius: ${({ theme: r }) => r.borderRadius} ${({ theme: r }) => r.borderRadius} 0 0;
    & > * {
      border-radius: ${({ theme: r }) => r.borderRadius} ${({ theme: r }) => r.borderRadius} 0 0;
    }

    &:hover {
      border-top: 1px solid ${({ theme: r }) => r.colors.primary600};
    }
  }

  & [data-strapi-expanded='true'] {
    border: 1px solid ${({ theme: r }) => r.colors.primary600};
  }

  ${({ theme: r, footer: e }) => `
    &:not(${e}) {
      & > *:last-of-type {
        border-radius: 0 0 ${r.borderRadius} ${r.borderRadius};
      }
    }
  `}
`, y = a(d)`
  svg path {
    fill: ${({ theme: r }) => r.colors.neutral500};
  }
`, p = ({ children: r, footer: e, label: l, labelAction: i, error: s }) => {
  const u = b.toArray(r).map((c) => m(c, { hasErrorMessage: !1 }));
  return /* @__PURE__ */ o.createElement(f, {
    attributeName: "data-strapi-accordion-toggle"
  }, l && /* @__PURE__ */ o.createElement($, {
    paddingBottom: 1
  }, /* @__PURE__ */ o.createElement(n, {
    variant: "pi",
    as: "label",
    textColor: "neutral800",
    fontWeight: "bold"
  }, l), i && /* @__PURE__ */ o.createElement(y, {
    paddingLeft: 1
  }, i)), /* @__PURE__ */ o.createElement(x, {
    footer: e
  }, u), e && /* @__PURE__ */ o.createElement(g, null, e), s && /* @__PURE__ */ o.createElement(d, {
    paddingTop: 1
  }, /* @__PURE__ */ o.createElement(n, {
    variant: "pi",
    textColor: "danger600"
  }, s)));
};
p.defaultProps = {
  footer: null,
  error: void 0,
  label: null,
  labelAction: void 0
};
p.propTypes = {
  children: t.node.isRequired,
  error: t.string,
  footer: t.node,
  label: t.string,
  labelAction: t.node
};
export {
  p as AccordionGroup
};
