// @nuxtjs/strapi/dist/module.mjs
import { defu } from 'defu';
import { defineNuxtModule, createResolver, addPlugin, addImportsDir, extendViteConfig } from '@nuxt/kit';

const module = defineNuxtModule({
  meta: {
    name: "@nuxtjs/strapi",
    configKey: "strapi",
    compatibility: {
      nuxt: "^3.0.0-rc.8"
    }
  },
  defaults: {
    url: process.env.STRAPI_URL || "http://localhost:1337",
    prefix: "/api",
    version: "v4",
    cookie: {},
    auth: {},
    cookieName: "strapi_jwt"
  },
  setup(options, nuxt) {
    if (!options.url) {
      throw new Error("Missing `STRAPI_URL` in `.env`");
    }
    nuxt.options.runtimeConfig.public.strapi = defu(nuxt.options.runtimeConfig.public.strapi, options);
    nuxt.options.runtimeConfig.strapi = defu(nuxt.options.runtimeConfig.strapi, options);
    const { resolve } = createResolver(import.meta.url);
    const runtimeDir = resolve("./runtime");
    nuxt.options.build.transpile.push(runtimeDir);
    addPlugin(resolve(runtimeDir, "strapi.plugin"));
    addImportsDir(resolve(runtimeDir, "composables"));
    addImportsDir(resolve(runtimeDir, `composables-${options.version}`));
    extendViteConfig((config) => {
      config.optimizeDeps = config.optimizeDeps || {};
      config.optimizeDeps.include = config.optimizeDeps.include || [];
      config.optimizeDeps.include.push("qs");
    });
  }
});

export { module as default };
