// @strapi/design-system/dist/MainNav/NavCondense.cjs
"use strict";var p=Object.defineProperty;var s=Object.getOwnPropertySymbols;var l=Object.prototype.hasOwnProperty,u=Object.prototype.propertyIsEnumerable;var d=(e,t,n)=>t in e?p(e,t,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[t]=n,c=(e,t)=>{for(var n in t||(t={}))l.call(t,n)&&d(e,n,t[n]);if(s)for(var n of s(t))u.call(t,n)&&d(e,n,t[n]);return e};var f=(e,t)=>{var n={};for(var o in e)l.call(e,o)&&t.indexOf(o)<0&&(n[o]=e[o]);if(e!=null&&s)for(var o of s(e))t.indexOf(o)<0&&u.call(e,o)&&(n[o]=e[o]);return n};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const v=require("react"),g=require("prop-types"),y=require("styled-components"),m=require("@strapi/icons/ChevronRight"),_=require("@strapi/icons/ChevronLeft"),b=require("../Icon/Icon.cjs"),C=require("../VisuallyHidden/VisuallyHidden.cjs"),$=require("./MainNavContext.cjs"),r=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},a=r(v),q=r(g),R=r(y),j=r(m),N=r(_),x=R.default.button`
  background: ${({theme:e})=>e.colors.neutral0};
  border: 1px solid ${({theme:e})=>e.colors.neutral150};
  border-radius: ${({theme:e})=>e.borderRadius};
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  bottom: ${(9+4)/16}rem; // 9 is the height of the svg and 4 is the padding below
  right: ${({theme:e,condensed:t})=>t?0:e.spaces[5]};
  transform: ${({condensed:e})=>e?"translateX(50%)":void 0};
  z-index: 2;
  width: ${18/16}rem;
  height: ${25/16}rem;

  svg {
    width: ${6/16}rem;
    height: ${9/16}rem;
  }
`,h=n=>{var o=n,{children:e}=o,t=f(o,["children"]);const i=$.useMainNav();return a.default.createElement(x,c({as:"button",condensed:i},t),a.default.createElement(b.Icon,{as:i?j.default:N.default,"aria-hidden":!0,color:"neutral600"}),a.default.createElement(C.VisuallyHidden,null,e))};h.propTypes={children:q.default.string.isRequired};exports.NavCondense=h;
