// @strapi/design-system/dist/MainNav/NavSections.js
var f = Object.defineProperty;
var p = Object.getOwnPropertySymbols;
var a = Object.prototype.hasOwnProperty, m = Object.prototype.propertyIsEnumerable;
var n = (e, r, t) => r in e ? f(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, d = (e, r) => {
  for (var t in r || (r = {}))
    a.call(r, t) && n(e, t, r[t]);
  if (p)
    for (var t of p(r))
      m.call(r, t) && n(e, t, r[t]);
  return e;
};
var c = (e, r) => {
  var t = {};
  for (var o in e)
    a.call(e, o) && r.indexOf(o) < 0 && (t[o] = e[o]);
  if (e != null && p)
    for (var o of p(e))
      r.indexOf(o) < 0 && m.call(e, o) && (t[o] = e[o]);
  return t;
};
import i, { Children as g } from "react";
import u from "prop-types";
import { Stack as h } from "../Stack/Stack.js";
import { Box as y } from "../Box/Box.js";
const E = (t) => {
  var o = t, { children: e } = o, r = c(o, ["children"]);
  return /* @__PURE__ */ i.createElement(y, {
    paddingLeft: 3,
    paddingRight: 2,
    paddingTop: 3
  }, /* @__PURE__ */ i.createElement(h, d({
    as: "ul",
    spacing: 4
  }, r), g.map(e, (l, s) => /* @__PURE__ */ i.createElement("li", {
    key: s
  }, l))));
};
E.propTypes = {
  children: u.node.isRequired
};
export {
  E as NavSections
};
