// @strapi/design-system/dist/Button/Button.cjs
"use strict";var W=Object.defineProperty;var i=Object.getOwnPropertySymbols;var m=Object.prototype.hasOwnProperty,B=Object.prototype.propertyIsEnumerable;var b=(e,t,a)=>t in e?W(e,t,{enumerable:!0,configurable:!0,writable:!0,value:a}):e[t]=a,v=(e,t)=>{for(var a in t||(t={}))m.call(t,a)&&b(e,a,t[a]);if(i)for(var a of i(t))B.call(t,a)&&b(e,a,t[a]);return e};var $=(e,t)=>{var a={};for(var o in e)m.call(e,o)&&t.indexOf(o)<0&&(a[o]=e[o]);if(e!=null&&i)for(var o of i(e))t.indexOf(o)<0&&B.call(e,o)&&(a[o]=e[o]);return a};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const k=require("react"),D=require("prop-types"),T=require("styled-components"),I=require("@strapi/icons/Loader"),_=require("../Typography/Typography.cjs"),f=require("../Box/Box.cjs"),r=require("./utils.cjs"),S=require("./constants.cjs"),N=require("../BaseButton/BaseButton.cjs"),s=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},n=s(k),l=s(D),p=s(T),O=s(I),w=T.keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`,A=p.default.div`
  animation: ${w} 2s infinite linear;
  will-change: transform;
`,C=p.default(f.Box)`
  height: 100%;
`,q=p.default(N.BaseButton)`
  align-items: center;
  background-color: ${({theme:e})=>e.colors.buttonPrimary600};
  border: 1px solid ${({theme:e})=>e.colors.buttonPrimary600};
  height: ${({theme:e,size:t})=>e.sizes.button[t]};
  padding-left: ${({theme:e})=>e.spaces[4]};
  padding-right: ${({theme:e})=>e.spaces[4]};

  ${f.Box} {
    display: flex;
    align-items: center;
  }

  ${_.Typography} {
    color: ${({theme:e})=>e.colors.buttonNeutral0};
  }

  &[aria-disabled='true'] {
    ${r.getDisabledStyle}
    &:active {
      ${r.getDisabledStyle}
    }
  }
  &:hover {
    ${r.getHoverStyle}
  }
  &:active {
    ${r.getActiveStyle}
  }
  ${r.getVariantStyle}
  ${({fullWidth:e})=>e&&`
    display: inline-flex;
    justify-content: center;
    width: 100%;
  `}
`,d=n.default.forwardRef((H,L)=>{var h=H,{variant:e,startIcon:t,endIcon:a,disabled:o,children:x,onClick:y,size:g,loading:u,fullWidth:j}=h,E=$(h,["variant","startIcon","endIcon","disabled","children","onClick","size","loading","fullWidth"]);const c=o||u,P=R=>{!c&&y&&y(R)};return n.default.createElement(q,v({ref:L,"aria-disabled":c,disabled:c,size:g,variant:e,onClick:P,fullWidth:j},E),(t||u)&&n.default.createElement(C,{"aria-hidden":!0,paddingRight:2},u?n.default.createElement(A,null,n.default.createElement(O.default,null)):t),n.default.createElement(_.Typography,{variant:g==="S"?"pi":void 0,fontWeight:"bold",lineHeight:0},x),a&&n.default.createElement(f.Box,{"aria-hidden":!0,paddingLeft:2},a))});d.displayName="Button";d.defaultProps={disabled:!1,endIcon:void 0,fullWidth:!1,loading:!1,onClick:void 0,size:"S",startIcon:void 0,variant:"default"};d.propTypes={children:l.default.node.isRequired,disabled:l.default.bool,endIcon:l.default.element,fullWidth:l.default.bool,loading:l.default.bool,onClick:l.default.func,size:l.default.oneOf(S.BUTTON_SIZES),startIcon:l.default.element,variant:l.default.oneOf(S.VARIANTS)};exports.Button=d;exports.ButtonWrapper=q;
