// @strapi/design-system/dist/Table/Table.cjs
"use strict";var _=Object.defineProperty;var s=Object.getOwnPropertySymbols;var w=Object.prototype.hasOwnProperty,T=Object.prototype.propertyIsEnumerable;var g=(e,o,t)=>o in e?_(e,o,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[o]=t,x=(e,o)=>{for(var t in o||(o={}))w.call(o,t)&&g(e,t,o[t]);if(s)for(var t of s(o))T.call(o,t)&&g(e,t,o[t]);return e};var y=(e,o)=>{var t={};for(var a in e)w.call(e,a)&&o.indexOf(a)<0&&(t[a]=e[a]);if(e!=null&&s)for(var a of s(e))o.indexOf(a)<0&&T.call(e,a)&&(t[a]=e[a]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const d=require("react"),v=require("prop-types"),k=require("styled-components"),q=require("../RawTable/RawTable.cjs"),f=require("../Box/Box.cjs"),h=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},n=h(d),u=h(v),c=h(k),B=c.default(f.Box)`
  overflow: hidden;
  border: 1px solid ${({theme:e})=>e.colors.neutral150};
`,$=c.default(q.RawTable)`
  width: 100%;
  white-space: nowrap;
`,L=c.default(f.Box)`
  position: relative;

  &:before {
    // TODO: make sure to add a token for this weird stuff
    background: linear-gradient(90deg, #c0c0cf 0%, rgba(0, 0, 0, 0) 100%);
    opacity: 0.2;
    position: absolute;
    height: 100%;
    content: ${({overflowing:e})=>e==="both"||e==="left"?"''":void 0};
    box-shadow: ${({theme:e})=>e.shadows.tableShadow};
    width: ${({theme:e})=>e.spaces[2]};
    left: 0;
  }

  &:after {
    // TODO: make sure to add a token for this weird stuff
    background: linear-gradient(270deg, #c0c0cf 0%, rgba(0, 0, 0, 0) 100%);
    opacity: 0.2;
    position: absolute;
    height: 100%;
    content: ${({overflowing:e})=>e==="both"||e==="right"?"''":void 0};
    box-shadow: ${({theme:e})=>e.shadows.tableShadow};
    width: ${({theme:e})=>e.spaces[2]};
    right: 0;
    top: 0;
  }
`,O=c.default(f.Box)`
  overflow-x: auto;
`,b=E=>{var p=E,{colCount:e,rowCount:o,footer:t}=p,a=y(p,["colCount","rowCount","footer"]);const i=d.useRef(null),[R,l]=d.useState(),m=r=>{const S=r.target.scrollWidth-r.target.clientWidth;if(r.target.scrollLeft===0){l("right");return}if(r.target.scrollLeft===S){l("left");return}r.target.scrollLeft>0&&l("both")};return d.useEffect(()=>{i.current.scrollWidth>i.current.clientWidth&&l("right")},[]),n.default.createElement(B,{shadow:"tableShadow",hasRadius:!0,background:"neutral0"},n.default.createElement(L,{overflowing:R},n.default.createElement(O,{ref:i,onScroll:m,paddingLeft:6,paddingRight:6},n.default.createElement($,x({colCount:e,rowCount:o},a)))),t)};b.defaultProps={footer:void 0};b.propTypes={colCount:u.default.number.isRequired,footer:u.default.node,rowCount:u.default.number.isRequired};exports.Table=b;
