// @strapi/design-system/dist/Table/TFooter.cjs
"use strict";var h=Object.defineProperty;var i=Object.getOwnPropertySymbols;var s=Object.prototype.hasOwnProperty,c=Object.prototype.propertyIsEnumerable;var u=(e,r,t)=>r in e?h(e,r,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[r]=t,p=(e,r)=>{for(var t in r||(r={}))s.call(r,t)&&u(e,t,r[t]);if(i)for(var t of i(r))c.call(r,t)&&u(e,t,r[t]);return e};var y=(e,r)=>{var t={};for(var o in e)s.call(e,o)&&r.indexOf(o)<0&&(t[o]=e[o]);if(e!=null&&i)for(var o of i(e))r.indexOf(o)<0&&c.call(e,o)&&(t[o]=e[o]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const x=require("react"),b=require("prop-types"),v=require("styled-components"),d=require("../Box/Box.cjs"),T=require("../Divider/Divider.cjs"),q=require("../Typography/Typography.cjs"),B=require("../Flex/Flex.cjs"),l=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},a=l(x),f=l(b),g=l(v),_=g.default(d.Box)`
  height: ${24/16}rem;
  width: ${24/16}rem;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;

  svg {
    height: ${10/16}rem;
    width: ${10/16}rem;
  }

  svg path {
    fill: ${({theme:e})=>e.colors.primary600};
  }
`,j=g.default(d.Box)`
  border-radius: 0 0 ${({theme:e})=>e.borderRadius} ${({theme:e})=>e.borderRadius};
  display: block;
  width: 100%;
  border: none;
`,m=o=>{var n=o,{children:e,icon:r}=n,t=y(n,["children","icon"]);return a.default.createElement("div",null,a.default.createElement(T.Divider,null),a.default.createElement(j,p({as:"button",background:"primary100",padding:5},t),a.default.createElement(B.Flex,null,a.default.createElement(_,{"aria-hidden":!0,background:"primary200"},r),a.default.createElement(d.Box,{paddingLeft:3},a.default.createElement(q.Typography,{variant:"pi",fontWeight:"bold",textColor:"primary600"},e)))))};m.propTypes={children:f.default.string.isRequired,icon:f.default.node.isRequired};exports.TFooter=m;
