// @strapi/design-system/dist/Layout/HeaderLayout.js
var B = Object.defineProperty, H = Object.defineProperties;
var T = Object.getOwnPropertyDescriptors;
var p = Object.getOwnPropertySymbols;
var E = Object.prototype.hasOwnProperty, v = Object.prototype.propertyIsEnumerable;
var h = (e, n, t) => n in e ? B(e, n, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[n] = t, l = (e, n) => {
  for (var t in n || (n = {}))
    E.call(n, t) && h(e, t, n[t]);
  if (p)
    for (var t of p(n))
      v.call(n, t) && h(e, t, n[t]);
  return e;
}, b = (e, n) => H(e, T(n));
var R = (e, n) => {
  var t = {};
  for (var a in e)
    E.call(e, a) && n.indexOf(a) < 0 && (t[a] = e[a]);
  if (e != null && p)
    for (var a of p(e))
      n.indexOf(a) < 0 && v.call(e, a) && (t[a] = e[a]);
  return t;
};
import r, { useRef as k, useState as C, useEffect as S } from "react";
import O from "styled-components";
import i from "prop-types";
import { Typography as u } from "../Typography/Typography.js";
import { Box as d } from "../Box/Box.js";
import { Flex as s } from "../Flex/Flex.js";
import { useElementOnScreen as A } from "../helpers/useElementOnScreen.js";
import { useResizeObserver as P } from "../helpers/useResizeObserver.js";
const j = () => {
  const e = k(null), [n, t] = C(null), [a, o] = A({
    root: null,
    rootMargin: "0px",
    threshold: 0
  });
  return P(a, () => {
    a.current && t(a.current.getBoundingClientRect());
  }), S(() => {
    e.current && t(e.current.getBoundingClientRect());
  }, [e]), {
    containerRef: a,
    isVisible: o,
    baseHeaderLayoutRef: e,
    headerSize: n
  };
}, m = (e) => {
  const { containerRef: n, isVisible: t, baseHeaderLayoutRef: a, headerSize: o } = j();
  return /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement("div", {
    style: { height: o == null ? void 0 : o.height },
    ref: n
  }, t && /* @__PURE__ */ r.createElement(c, l({
    ref: a
  }, e))), !t && /* @__PURE__ */ r.createElement(c, b(l({}, e), {
    sticky: !0,
    width: o == null ? void 0 : o.width
  })));
};
m.displayName = "HeaderLayout";
const q = O(d)`
  position: fixed;
  top: 0;
  right: 0;
  width: ${(e) => e.width}px;
  z-index: 4;
  box-shadow: ${({ theme: e }) => e.shadows.tableShadow};
`, c = r.forwardRef(
  (z, L) => {
    var g = z, { navigationAction: e, primaryAction: n, secondaryAction: t, subtitle: a, title: o, sticky: w, width: x } = g, f = R(g, ["navigationAction", "primaryAction", "secondaryAction", "subtitle", "title", "sticky", "width"]);
    const y = typeof a == "string";
    return w ? /* @__PURE__ */ r.createElement(q, {
      paddingLeft: 6,
      paddingRight: 6,
      paddingTop: 3,
      paddingBottom: 3,
      background: "neutral0",
      width: x,
      "data-strapi-header-sticky": !0
    }, /* @__PURE__ */ r.createElement(s, {
      justifyContent: "space-between"
    }, /* @__PURE__ */ r.createElement(s, null, e && /* @__PURE__ */ r.createElement(d, {
      paddingRight: 3
    }, e), /* @__PURE__ */ r.createElement(d, null, /* @__PURE__ */ r.createElement(u, l({
      variant: "beta",
      as: "h1"
    }, f), o), y ? /* @__PURE__ */ r.createElement(u, {
      variant: "pi",
      textColor: "neutral600"
    }, a) : a), t ? /* @__PURE__ */ r.createElement(d, {
      paddingLeft: 4
    }, t) : null), /* @__PURE__ */ r.createElement(s, null, n ? /* @__PURE__ */ r.createElement(d, {
      paddingLeft: 2
    }, n) : void 0))) : /* @__PURE__ */ r.createElement(d, {
      ref: L,
      paddingLeft: 10,
      paddingRight: 10,
      paddingBottom: 8,
      paddingTop: e ? 6 : 8,
      background: "neutral100",
      "data-strapi-header": !0
    }, e ? /* @__PURE__ */ r.createElement(d, {
      paddingBottom: 2
    }, e) : null, /* @__PURE__ */ r.createElement(s, {
      justifyContent: "space-between"
    }, /* @__PURE__ */ r.createElement(s, null, /* @__PURE__ */ r.createElement(u, l({
      as: "h1",
      variant: "alpha"
    }, f), o), t ? /* @__PURE__ */ r.createElement(d, {
      paddingLeft: 4
    }, t) : null), n), y ? /* @__PURE__ */ r.createElement(u, {
      variant: "epsilon",
      textColor: "neutral600",
      as: "p"
    }, a) : a);
  }
);
c.displayName = "BaseHeaderLayout";
c.defaultProps = {
  navigationAction: void 0,
  primaryAction: void 0,
  secondaryAction: void 0,
  subtitle: void 0,
  sticky: !1,
  width: void 0
};
c.propTypes = {
  navigationAction: i.node,
  primaryAction: i.node,
  secondaryAction: i.node,
  sticky: i.bool,
  subtitle: i.oneOfType([i.string, i.node]),
  title: i.string.isRequired,
  width: i.number
};
m.defaultProps = {
  navigationAction: void 0,
  primaryAction: void 0,
  secondaryAction: void 0,
  subtitle: void 0
};
m.propTypes = {
  navigationAction: i.node,
  primaryAction: i.node,
  secondaryAction: i.node,
  subtitle: i.oneOfType([i.string, i.node]),
  title: i.string.isRequired
};
export {
  c as BaseHeaderLayout,
  m as HeaderLayout
};
