// @strapi/design-system/dist/SubNav/SubNavHeader.js
import e, { useState as w, useRef as d, useEffect as x } from "react";
import r from "prop-types";
import D from "styled-components";
import I from "@strapi/icons/Search";
import { Flex as K } from "../Flex/Flex.js";
import { Typography as P } from "../Typography/Typography.js";
import { IconButton as k } from "../IconButton/IconButton.js";
import { Box as n } from "../Box/Box.js";
import { Divider as O } from "../Divider/Divider.js";
import { Searchbar as F } from "../Searchbar/Searchbar.js";
import { SearchForm as _ } from "../Searchbar/SearchForm.js";
import { useId as $ } from "../helpers/useId.js";
import { usePrevious as j } from "../helpers/usePrevious.js";
import { KeyboardKeys as q } from "../helpers/keyboardKeys.js";
const u = D(O)`
  width: ${24 / 16}rem;
  background-color: ${({ theme: l }) => l.colors.neutral200};
`, f = ({ as: l, label: g, searchLabel: s, searchable: h, onChange: b, value: E, onClear: S, onSubmit: y, id: C }) => {
  const [a, c] = w(!1), m = j(a), v = $("subnav-searchbar-clear", C), o = d(), i = d();
  x(() => {
    a && o.current && o.current.focus(), m && !a && i.current && i.current.focus();
  }, [a, m]);
  const B = () => {
    c((t) => !t);
  }, T = (t) => {
    S(t), o.current.focus();
  }, R = (t) => {
    var p;
    ((p = t.relatedTarget) == null ? void 0 : p.id) !== v && c(!1);
  }, L = (t) => {
    t.key === q.ESCAPE && c(!1);
  };
  return a ? /* @__PURE__ */ e.createElement(n, {
    paddingLeft: 4,
    paddingTop: 5,
    paddingBottom: 2,
    paddingRight: 4
  }, /* @__PURE__ */ e.createElement(_, null, /* @__PURE__ */ e.createElement(F, {
    name: "searchbar",
    value: E,
    onChange: b,
    placeholder: "e.g: strapi-plugin-abcd",
    onKeyDown: L,
    ref: o,
    onBlur: R,
    onClear: T,
    onSubmit: y,
    clearLabel: "Clear",
    size: "S"
  }, s)), /* @__PURE__ */ e.createElement(n, {
    paddingLeft: 2,
    paddingTop: 4
  }, /* @__PURE__ */ e.createElement(u, null))) : /* @__PURE__ */ e.createElement(n, {
    paddingLeft: 6,
    paddingTop: 6,
    paddingBottom: 2,
    paddingRight: 4
  }, /* @__PURE__ */ e.createElement(K, {
    justifyContent: "space-between",
    alignItems: "flex-start"
  }, /* @__PURE__ */ e.createElement(P, {
    variant: "beta",
    as: l
  }, g), h && /* @__PURE__ */ e.createElement(k, {
    ref: i,
    onClick: B,
    label: s,
    icon: /* @__PURE__ */ e.createElement(I, null)
  })), /* @__PURE__ */ e.createElement(n, {
    paddingTop: 4
  }, /* @__PURE__ */ e.createElement(u, null)));
};
f.defaultProps = {
  as: "h2",
  searchable: !1,
  onChange() {
  },
  onClear() {
  },
  onSubmit() {
  },
  value: "",
  searchLabel: "",
  id: void 0
};
f.propTypes = {
  as: r.string,
  id: r.string,
  label: r.string.isRequired,
  onChange: r.func,
  onClear: r.func,
  onSubmit: r.func,
  searchLabel: r.string,
  searchable: r.bool,
  value: r.string
};
export {
  f as SubNavHeader
};
