// @strapi/design-system/dist/ProgressBar/ProgressBar.js
var f = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var m = Object.prototype.hasOwnProperty, l = Object.prototype.propertyIsEnumerable;
var u = (r, o, e) => o in r ? f(r, o, { enumerable: !0, configurable: !0, writable: !0, value: e }) : r[o] = e, p = (r, o) => {
  for (var e in o || (o = {}))
    m.call(o, e) && u(r, e, o[e]);
  if (i)
    for (var e of i(o))
      l.call(o, e) && u(r, e, o[e]);
  return r;
};
var d = (r, o) => {
  var e = {};
  for (var a in r)
    m.call(r, a) && o.indexOf(a) < 0 && (e[a] = r[a]);
  if (r != null && i)
    for (var a of i(r))
      o.indexOf(a) < 0 && l.call(r, a) && (e[a] = r[a]);
  return e;
};
import x from "react";
import t from "prop-types";
import g from "styled-components";
import { Box as h } from "../Box/Box.js";
const v = g(h)`
  &:before {
    background-color: ${({ theme: r }) => r.colors.neutral0};
    border-radius: ${({ theme: r }) => r.borderRadius};
    bottom: 0;
    content: '';
    position: absolute;
    top: 0;
    width: ${({ value: r }) => `${r}%`};
  }
`, b = (P) => {
  var n = P, { min: r, max: o, value: e, children: a, size: s } = n, c = d(n, ["min", "max", "value", "children", "size"]);
  return /* @__PURE__ */ x.createElement(v, p({
    background: "neutral600",
    hasRadius: !0,
    "aria-label": a,
    "aria-valuemax": o,
    "aria-valuemin": r,
    "aria-valuenow": e,
    height: s === "S" ? 1 : 2,
    position: "relative",
    role: "progressbar",
    value: e,
    width: s === "S" ? "78px" : "102px"
  }, c));
};
b.defaultProps = {
  min: 0,
  max: 100,
  value: 0,
  size: "M"
};
b.propTypes = {
  children: t.node.isRequired,
  max: t.number,
  min: t.number,
  size: t.oneOf(["S", "M"]),
  value: t.number
};
export {
  b as ProgressBar
};
