// nuxt-swiper/dist/module.mjs
import * as swiper from 'swiper';
import { defineNuxtModule, useLogger, isNodeModules, addImports, addPluginTemplate } from '@nuxt/kit';

const name = "nuxt-swiper";
const version = "0.1.6";

const module = defineNuxtModule({
  meta: {
    name,
    version,
    configKey: "swiper"
  },
  defaults: {
    prefix: "Swiper",
    styleLang: "css",
    modules: "*"
  },
  setup(_options, nuxt) {
    let { styleLang } = _options;
    const logger = useLogger(name);
    const { prefix, modules } = _options;
    const cssImports = [`import 'swiper/${styleLang}'`];
    const moduleImports = [
      {
        name: "useSwiper",
        as: "useSwiper",
        from: "swiper/vue"
      },
      {
        name: "useSwiperSlide",
        as: "useSwiperSlide",
        from: "swiper/vue"
      }
    ];
    if (styleLang === "scss" && !isNodeModules("sass")) {
      styleLang = "css";
      logger.warn(
        "[nuxt-swiper]: You need to install `sass` to use the scss option. Falling back to `css`."
      );
    }
    for (const [key, _] of Object.entries(swiper)) {
      const snakeCase = key.replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, "$1-$2").replace(/^-/, "").toLowerCase();
      const hasModule = modules === "*" || Array.isArray(modules) && modules.includes(snakeCase);
      if (hasModule && !["default", "Swiper"].includes(key)) {
        const noCssInModules = [
          "autoplay",
          "controller",
          "effect-coverflow",
          "hash-navigation",
          "history",
          "keyboard",
          "manipulation",
          "mousewheel",
          "parallax",
          "thumbs"
        ];
        if (!noCssInModules.includes(snakeCase)) {
          cssImports.push(`import 'swiper/${styleLang}/${snakeCase}'`);
        }
        moduleImports.push({
          name: key,
          as: `${prefix}${key}`,
          from: "swiper"
        });
      }
    }
    addImports(moduleImports);
    addPluginTemplate({
      filename: "swiper.mjs",
      getContents() {
        const lines = [
          "import { Swiper, SwiperSlide } from 'swiper/vue'",
          `export default defineNuxtPlugin((nuxtApp) => {
            nuxtApp.vueApp.component('Swiper', Swiper)
            nuxtApp.vueApp.component('SwiperSlide', SwiperSlide)
          })`
        ];
        lines.unshift(...cssImports);
        return lines.join("\n");
      }
    });
  }
});

export { module as default };
