// @strapi/design-system/dist/Card/CardTimer.js
var l = Object.defineProperty;
var a = Object.getOwnPropertySymbols;
var i = Object.prototype.hasOwnProperty, m = Object.prototype.propertyIsEnumerable;
var p = (r, e, o) => e in r ? l(r, e, { enumerable: !0, configurable: !0, writable: !0, value: o }) : r[e] = o, s = (r, e) => {
  for (var o in e || (e = {}))
    i.call(e, o) && p(r, o, e[o]);
  if (a)
    for (var o of a(e))
      m.call(e, o) && p(r, o, e[o]);
  return r;
};
var n = (r, e) => {
  var o = {};
  for (var t in r)
    i.call(r, t) && e.indexOf(t) < 0 && (o[t] = r[t]);
  if (r != null && a)
    for (var t of a(r))
      e.indexOf(t) < 0 && m.call(r, t) && (o[t] = r[t]);
  return o;
};
import d from "react";
import c from "prop-types";
import u from "styled-components";
import { Box as f } from "../Box/Box.js";
import { Typography as y } from "../Typography/Typography.js";
const T = u(f)`
  position: absolute;
  bottom: ${({ theme: r }) => r.spaces[1]};
  right: ${({ theme: r }) => r.spaces[1]};
`, g = (o) => {
  var t = o, { children: r } = t, e = n(t, ["children"]);
  return /* @__PURE__ */ d.createElement(T, s({
    padding: 1,
    background: "neutral800",
    color: "neutral0",
    as: "time",
    hasRadius: !0
  }, e), /* @__PURE__ */ d.createElement(y, {
    variant: "pi",
    textColor: "neutral0"
  }, r));
};
g.propTypes = {
  children: c.node.isRequired
};
export {
  g as CardTimer
};
