// @strapi/design-system/dist/Card/Card.js
var b = Object.defineProperty;
var a = Object.getOwnPropertySymbols;
var l = Object.prototype.hasOwnProperty, p = Object.prototype.propertyIsEnumerable;
var i = (r, e, t) => e in r ? b(r, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : r[e] = t, m = (r, e) => {
  for (var t in e || (e = {}))
    l.call(e, t) && i(r, t, e[t]);
  if (a)
    for (var t of a(e))
      p.call(e, t) && i(r, t, e[t]);
  return r;
};
var n = (r, e) => {
  var t = {};
  for (var o in r)
    l.call(r, o) && e.indexOf(o) < 0 && (t[o] = r[o]);
  if (r != null && a)
    for (var o of a(r))
      e.indexOf(o) < 0 && p.call(r, o) && (t[o] = r[o]);
  return t;
};
import s from "react";
import c from "prop-types";
import { Box as f } from "../Box/Box.js";
import { CardContext as x } from "./CardContext.js";
import { useId as h } from "../helpers/useId.js";
const u = (t) => {
  var o = t, { id: r } = o, e = n(o, ["id"]);
  const d = h("card", r);
  return /* @__PURE__ */ s.createElement(x.Provider, {
    value: { id: d }
  }, /* @__PURE__ */ s.createElement(f, m({
    id: r,
    tabIndex: 0,
    hasRadius: !0,
    background: "neutral0",
    borderStyle: "solid",
    borderWidth: "1px",
    borderColor: "neutral150",
    shadow: "tableShadow",
    as: "article",
    "aria-labelledby": `${d}-title`
  }, e)));
};
u.defaultProps = {
  id: void 0
};
u.propTypes = {
  id: c.string
};
export {
  u as Card
};
