// @strapi/design-system/dist/Radio/Radio.js
var f = Object.defineProperty;
var a = Object.getOwnPropertySymbols;
var p = Object.prototype.hasOwnProperty, l = Object.prototype.propertyIsEnumerable;
var m = (e, r, t) => r in e ? f(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, n = (e, r) => {
  for (var t in r || (r = {}))
    p.call(r, t) && m(e, t, r[t]);
  if (a)
    for (var t of a(r))
      l.call(r, t) && m(e, t, r[t]);
  return e;
};
var d = (e, r) => {
  var t = {};
  for (var o in e)
    p.call(e, o) && r.indexOf(o) < 0 && (t[o] = e[o]);
  if (e != null && a)
    for (var o of a(e))
      r.indexOf(o) < 0 && l.call(e, o) && (t[o] = e[o]);
  return t;
};
import i from "react";
import s from "prop-types";
import c from "styled-components";
import { BaseRadio as y } from "../BaseRadio/BaseRadio.js";
import { Typography as u } from "../Typography/Typography.js";
import { Box as x } from "../Box/Box.js";
const R = c(u)`
  display: flex;
  align-items: center;
`, T = (t) => {
  var o = t, { children: e } = o, r = d(o, ["children"]);
  return /* @__PURE__ */ i.createElement(R, {
    as: "label",
    textColor: "neutral800"
  }, /* @__PURE__ */ i.createElement(y, n({}, r)), /* @__PURE__ */ i.createElement(x, {
    paddingLeft: 2
  }, e));
};
T.propTypes = {
  children: s.node.isRequired,
  value: s.any.isRequired
};
export {
  T as Radio
};
