// @strapi/design-system/dist/v2/MainNav/NavSection.js
var R = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var c = Object.prototype.hasOwnProperty, u = Object.prototype.propertyIsEnumerable;
var s = (e, r, t) => r in e ? R(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, l = (e, r) => {
  for (var t in r || (r = {}))
    c.call(r, t) && s(e, t, r[t]);
  if (i)
    for (var t of i(r))
      u.call(r, t) && s(e, t, r[t]);
  return e;
};
var g = (e, r) => {
  var t = {};
  for (var n in e)
    c.call(e, n) && r.indexOf(n) < 0 && (t[n] = e[n]);
  if (e != null && i)
    for (var n of i(e))
      r.indexOf(n) < 0 && u.call(e, n) && (t[n] = e[n]);
  return t;
};
import a, { Children as E } from "react";
import f from "prop-types";
import { Box as y } from "../../Box/Box.js";
import { Typography as h } from "../../Typography/Typography.js";
import { Stack as o } from "../../Stack/Stack.js";
import { useMainNav as k } from "./MainNavContext.js";
import { VisuallyHidden as T } from "../../VisuallyHidden/VisuallyHidden.js";
import { Divider as v } from "../../Divider/Divider.js";
const x = (n) => {
  var d = n, { label: e, children: r } = d, t = g(d, ["label", "children"]);
  return k() ? /* @__PURE__ */ a.createElement(o, {
    spacing: 2
  }, /* @__PURE__ */ a.createElement(y, {
    paddingTop: 1,
    paddingBottom: 1,
    background: "neutral0",
    hasRadius: !0,
    as: "span"
  }, /* @__PURE__ */ a.createElement(v, null), /* @__PURE__ */ a.createElement(T, null, /* @__PURE__ */ a.createElement("span", null, e))), /* @__PURE__ */ a.createElement(o, l({
    as: "ul",
    spacing: 2
  }, t), E.map(r, (m, p) => /* @__PURE__ */ a.createElement("li", {
    key: p
  }, m)))) : /* @__PURE__ */ a.createElement(o, {
    spacing: 2
  }, /* @__PURE__ */ a.createElement(y, {
    paddingTop: 1,
    paddingBottom: 1,
    background: "neutral0",
    paddingRight: 3,
    paddingLeft: 3,
    hasRadius: !0,
    as: "span"
  }, /* @__PURE__ */ a.createElement(h, {
    variant: "sigma",
    textColor: "neutral600"
  }, e)), /* @__PURE__ */ a.createElement(o, l({
    as: "ul",
    spacing: 2
  }, t), E.map(r, (m, p) => /* @__PURE__ */ a.createElement("li", {
    key: p
  }, m))));
};
x.propTypes = {
  children: f.node.isRequired,
  label: f.string.isRequired
};
export {
  x as NavSection
};
