// @strapi/design-system/dist/TextButton/TextButton.cjs
"use strict";var B=Object.defineProperty;var l=Object.getOwnPropertySymbols;var y=Object.prototype.hasOwnProperty,m=Object.prototype.propertyIsEnumerable;var p=(e,a,t)=>a in e?B(e,a,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[a]=t,g=(e,a)=>{for(var t in a||(a={}))y.call(a,t)&&p(e,t,a[t]);if(l)for(var t of l(a))m.call(a,t)&&p(e,t,a[t]);return e};var x=(e,a)=>{var t={};for(var o in e)y.call(e,o)&&a.indexOf(o)<0&&(t[o]=e[o]);if(e!=null&&l)for(var o of l(e))a.indexOf(o)<0&&m.call(e,o)&&(t[o]=e[o]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const j=require("react"),E=require("prop-types"),h=require("styled-components"),L=require("@strapi/icons/Loader"),b=require("../Box/Box.cjs"),k=require("../Typography/Typography.cjs"),F=require("../Flex/Flex.cjs"),R=require("../themes/utils.cjs"),s=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},r=s(j),n=s(E),v=s(h),$=s(L),C=h.keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`,P=v.default.div`
  animation: ${C} 2s infinite linear;
  will-change: transform;
`,S=v.default(F.Flex)`
  background: transparent;
  border: none;

  &[aria-disabled='true'] {
    pointer-events: none;
    svg path {
      fill: ${({theme:e})=>e.colors.neutral600};
    }
  }

  svg {
    display: flex;
    font-size: ${10/16}rem;
  }

  svg path {
    fill: ${({theme:e})=>e.colors.primary600};
  }

  ${R.buttonFocusStyle}
`,d=r.default.forwardRef((w,_)=>{var c=w,{children:e,startIcon:a,endIcon:t,onClick:o,disabled:u,loading:i}=c,T=x(c,["children","startIcon","endIcon","onClick","disabled","loading"]);const q=o&&!u?o:void 0,f=u||i;return r.default.createElement(S,g({ref:_,"aria-disabled":f,onClick:q,as:"button",type:"button"},T),(a||i)&&r.default.createElement(b.Box,{as:"span",paddingRight:2,"aria-hidden":!0},i?r.default.createElement(P,null,r.default.createElement($.default,null)):a),r.default.createElement(k.Typography,{variant:"pi",textColor:f?"neutral600":"primary600"},e),t&&r.default.createElement(b.Box,{as:"span",paddingLeft:2,"aria-hidden":!0},t))});d.displayName="TextButton";d.defaultProps={disabled:!1,startIcon:void 0,endIcon:void 0,loading:!1,onClick:void 0};d.propTypes={children:n.default.node.isRequired,disabled:n.default.bool,endIcon:n.default.element,loading:n.default.bool,onClick:n.default.func,startIcon:n.default.element};exports.TextButton=d;
