// @strapi/design-system/dist/v2/LinkButton/LinkButton.cjs
"use strict";var x=Object.defineProperty,q=Object.defineProperties;var _=Object.getOwnPropertyDescriptors;var d=Object.getOwnPropertySymbols;var b=Object.prototype.hasOwnProperty,g=Object.prototype.propertyIsEnumerable;var B=(e,t,a)=>t in e?x(e,t,{enumerable:!0,configurable:!0,writable:!0,value:a}):e[t]=a,v=(e,t)=>{for(var a in t||(t={}))b.call(t,a)&&B(e,a,t[a]);if(d)for(var a of d(t))g.call(t,a)&&B(e,a,t[a]);return e},h=(e,t)=>q(e,_(t));var m=(e,t)=>{var a={};for(var o in e)b.call(e,o)&&t.indexOf(o)<0&&(a[o]=e[o]);if(e!=null&&d)for(var o of d(e))t.indexOf(o)<0&&g.call(e,o)&&(a[o]=e[o]);return a};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const j=require("react"),E=require("styled-components"),R=require("prop-types"),l=require("../../Typography/Typography.cjs"),u=require("../../Box/Box.cjs"),i=require("../../Button/utils.cjs"),$=require("../../Button/constants.cjs"),P=require("../../BaseButton/BaseButton.cjs"),T=require("../../BaseLink/BaseLink.cjs"),c=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},n=c(j),I=c(E),r=c(R),N=I.default(P.BaseButtonWrapper)`
  padding: ${({theme:e,size:t})=>`${t==="S"?e.spaces[2]:"10px"} ${e.spaces[4]}`};
  background: ${({theme:e})=>e.colors.buttonPrimary600};
  border: 1px solid ${({theme:e})=>e.colors.buttonPrimary600};
  border-radius: ${({theme:e})=>e.borderRadius};
  ${u.Box} {
    display: flex;
    align-items: center;
  }
  ${l.Typography} {
    color: ${({theme:e})=>e.colors.buttonNeutral0};
  }
  &[aria-disabled='true'] {
    ${i.getDisabledStyle}
    &:active {
      ${i.getDisabledStyle}
    }
  }
  &:hover {
    ${i.getHoverStyle}
  }
  &:active {
    ${i.getActiveStyle}
  }
  ${i.getVariantStyle}

  /**
    Link specific properties
  */
  display: inline-flex;
  text-decoration: none;
  pointer-events: ${({disabled:e})=>e?"none":void 0};
`,s=n.default.forwardRef((O,L)=>{var y=O,{variant:e,startIcon:t,endIcon:a,disabled:o,children:f,size:p,as:k}=y,S=m(y,["variant","startIcon","endIcon","disabled","children","size","as"]);return n.default.createElement(N,h(v({ref:L,"aria-disabled":o,size:p,variant:e},S),{as:k||T.BaseLink}),t&&n.default.createElement(u.Box,{"aria-hidden":!0,paddingRight:2},t),p==="S"?n.default.createElement(l.Typography,{variant:"pi",fontWeight:"bold"},f):n.default.createElement(l.Typography,{fontWeight:"bold"},f),a&&n.default.createElement(u.Box,{"aria-hidden":!0,paddingLeft:2},a))});s.displayName="LinkButton";s.defaultProps={as:T.BaseLink,disabled:!1,startIcon:void 0,endIcon:void 0,size:"S",variant:"default",onClick:void 0,href:void 0,to:void 0};s.propTypes={as:r.default.elementType,children:r.default.node.isRequired,disabled:r.default.bool,endIcon:r.default.element,href(e){if(!e.disabled&&!e.to&&!e.href)return new Error("href must be defined")},onClick:r.default.func,size:r.default.oneOf($.BUTTON_SIZES),startIcon:r.default.element,to(e){if(!e.disabled&&!e.href&&!e.to)return new Error("to must be defined")},variant:r.default.oneOf($.VARIANTS)};exports.LinkButton=s;
