// @strapi/design-system/dist/ModalLayout/ModalLayout.js
var y = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var m = Object.prototype.hasOwnProperty, p = Object.prototype.propertyIsEnumerable;
var n = (e, r, o) => r in e ? y(e, r, { enumerable: !0, configurable: !0, writable: !0, value: o }) : e[r] = o, s = (e, r) => {
  for (var o in r || (r = {}))
    m.call(r, o) && n(e, o, r[o]);
  if (i)
    for (var o of i(r))
      p.call(r, o) && n(e, o, r[o]);
  return e;
};
var d = (e, r) => {
  var o = {};
  for (var t in e)
    m.call(e, t) && r.indexOf(t) < 0 && (o[t] = e[t]);
  if (e != null && i)
    for (var t of i(e))
      r.indexOf(t) < 0 && p.call(e, t) && (o[t] = e[t]);
  return o;
};
import a from "react";
import u from "styled-components";
import c from "prop-types";
import { Box as g } from "../Box/Box.js";
import { FocusTrap as E } from "../FocusTrap/FocusTrap.js";
import { Portal as b } from "../Portal/Portal.js";
import { ModalContext as x } from "./ModalContext.js";
import { DismissibleLayer as w } from "../DismissibleLayer/DismissibleLayer.js";
import P from "../helpers/useLockScroll.js";
const h = u.div`
  position: fixed;
  z-index: 4;
  inset: 0;
  background: ${({ theme: e }) => `${e.colors.neutral800}1F`};
  padding: 0 ${({ theme: e }) => e.spaces[8]};
  display: flex;
  align-items: center;
  justify-content: center;
`, k = u(g)`
  width: ${830 / 16}rem;
`, M = (t) => {
  var l = t, { onClose: e, labelledBy: r } = l, o = d(l, ["onClose", "labelledBy"]);
  return P(!0), /* @__PURE__ */ a.createElement(b, null, /* @__PURE__ */ a.createElement(x.Provider, {
    value: e
  }, /* @__PURE__ */ a.createElement(h, null, /* @__PURE__ */ a.createElement(E, null, /* @__PURE__ */ a.createElement(w, {
    onEscapeKeyDown: e,
    onPointerDownOutside: e
  }, /* @__PURE__ */ a.createElement(k, s({
    "aria-labelledby": r,
    onClick: (f) => f.stopPropagation(),
    background: "neutral0",
    hasRadius: !0,
    shadow: "popupShadow",
    role: "dialog",
    "aria-modal": !0
  }, o)))))));
};
M.propTypes = {
  labelledBy: c.string.isRequired,
  onClose: c.func.isRequired
};
export {
  M as ModalLayout
};
