// @strapi/design-system/dist/LinkButton/LinkButton.cjs
"use strict";var E=Object.defineProperty,L=Object.defineProperties;var R=Object.getOwnPropertyDescriptors;var d=Object.getOwnPropertySymbols;var B=Object.prototype.hasOwnProperty,m=Object.prototype.propertyIsEnumerable;var b=(e,t,o)=>t in e?E(e,t,{enumerable:!0,configurable:!0,writable:!0,value:o}):e[t]=o,h=(e,t)=>{for(var o in t||(t={}))B.call(t,o)&&b(e,o,t[o]);if(d)for(var o of d(t))m.call(t,o)&&b(e,o,t[o]);return e},$=(e,t)=>L(e,R(t));var S=(e,t)=>{var o={};for(var r in e)B.call(e,r)&&t.indexOf(r)<0&&(o[r]=e[r]);if(e!=null&&d)for(var r of d(e))t.indexOf(r)<0&&m.call(e,r)&&(o[r]=e[r]);return o};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const j=require("react"),P=require("react-router-dom"),N=require("styled-components"),D=require("prop-types"),u=require("../Typography/Typography.cjs"),c=require("../Box/Box.cjs"),i=require("../Button/utils.cjs"),T=require("../Button/constants.cjs"),I=require("../BaseButton/BaseButton.cjs"),f=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},n=f(j),O=f(N),a=f(D),W=O.default(I.BaseButtonWrapper)`
  padding: ${({theme:e,size:t})=>`${t==="S"?e.spaces[2]:"10px"} ${e.spaces[4]}`};
  background: ${({theme:e})=>e.colors.buttonPrimary600};
  border: 1px solid ${({theme:e})=>e.colors.buttonPrimary600};
  border-radius: ${({theme:e})=>e.borderRadius};
  ${c.Box} {
    display: flex;
    align-items: center;
  }
  ${u.Typography} {
    color: ${({theme:e})=>e.colors.buttonNeutral0};
  }
  &[aria-disabled='true'] {
    ${i.getDisabledStyle}
    &:active {
      ${i.getDisabledStyle}
    }
  }
  &:hover {
    ${i.getHoverStyle}
  }
  &:active {
    ${i.getActiveStyle}
  }
  ${i.getVariantStyle}
  /**
    Link specific properties
  */
  display: inline-flex;
  text-decoration: none;
  pointer-events: ${({disabled:e})=>e?"none":void 0};
`,l=n.default.forwardRef((w,_)=>{var v=w,{variant:e,startIcon:t,endIcon:o,disabled:r,children:p,size:y,href:s,to:g}=v,x=S(v,["variant","startIcon","endIcon","disabled","children","size","href","to"]);const k=s?"_blank":void 0,q=s?"noreferrer noopener":void 0;return n.default.createElement(W,$(h({ref:_,"aria-disabled":r,size:y,variant:e,target:k,rel:q,to:r?void 0:g,href:r?"#":s},x),{as:g&&!r?P.NavLink:"a"}),t&&n.default.createElement(c.Box,{"aria-hidden":!0,paddingRight:2},t),y==="S"?n.default.createElement(u.Typography,{variant:"pi",fontWeight:"bold"},p):n.default.createElement(u.Typography,{fontWeight:"bold"},p),o&&n.default.createElement(c.Box,{"aria-hidden":!0,paddingLeft:2},o))});l.displayName="LinkButton";l.defaultProps={disabled:!1,startIcon:void 0,endIcon:void 0,size:"S",variant:"default",onClick:void 0,href:void 0,to:void 0};l.propTypes={children:a.default.node.isRequired,disabled:a.default.bool,endIcon:a.default.element,href(e){if(!e.disabled&&!e.to&&!e.href)return new Error("href must be defined")},onClick:a.default.func,size:a.default.oneOf(T.BUTTON_SIZES),startIcon:a.default.element,to(e){if(!e.disabled&&!e.href&&!e.to)return new Error("to must be defined")},variant:a.default.oneOf(T.VARIANTS)};exports.LinkButton=l;
