// @strapi/design-system/dist/v2/SubNav/SubNavSections.js
var f = Object.defineProperty;
var p = Object.getOwnPropertySymbols;
var n = Object.prototype.hasOwnProperty, a = Object.prototype.propertyIsEnumerable;
var m = (e, r, t) => r in e ? f(e, r, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[r] = t, c = (e, r) => {
  for (var t in r || (r = {}))
    n.call(r, t) && m(e, t, r[t]);
  if (p)
    for (var t of p(r))
      a.call(r, t) && m(e, t, r[t]);
  return e;
};
var d = (e, r) => {
  var t = {};
  for (var o in e)
    n.call(e, o) && r.indexOf(o) < 0 && (t[o] = e[o]);
  if (e != null && p)
    for (var o of p(e))
      r.indexOf(o) < 0 && a.call(e, o) && (t[o] = e[o]);
  return t;
};
import i, { Children as u } from "react";
import g from "prop-types";
import { Stack as y } from "../../Stack/Stack.js";
import { Box as E } from "../../Box/Box.js";
const S = (t) => {
  var o = t, { children: e } = o, r = d(o, ["children"]);
  return /* @__PURE__ */ i.createElement(E, {
    paddingTop: 2,
    paddingBottom: 4
  }, /* @__PURE__ */ i.createElement(y, c({
    as: "ol",
    spacing: 2
  }, r), u.map(e, (l, s) => /* @__PURE__ */ i.createElement("li", {
    key: s
  }, l))));
};
S.propTypes = {
  children: g.node.isRequired
};
export {
  S as SubNavSections
};
