// @strapi/design-system/dist/v2/SubNav/SubNav.js
var m = Object.defineProperty;
var i = Object.getOwnPropertySymbols;
var p = Object.prototype.hasOwnProperty, s = Object.prototype.propertyIsEnumerable;
var a = (r, o, t) => o in r ? m(r, o, { enumerable: !0, configurable: !0, writable: !0, value: t }) : r[o] = t, l = (r, o) => {
  for (var t in o || (o = {}))
    p.call(o, t) && a(r, t, o[t]);
  if (i)
    for (var t of i(o))
      s.call(o, t) && a(r, t, o[t]);
  return r;
};
var n = (r, o) => {
  var t = {};
  for (var e in r)
    p.call(r, e) && o.indexOf(e) < 0 && (t[e] = r[e]);
  if (r != null && i)
    for (var e of i(r))
      o.indexOf(e) < 0 && s.call(r, e) && (t[e] = r[e]);
  return t;
};
import d from "react";
import u from "styled-components";
import c from "prop-types";
import { Grid as b } from "../../Grid/Grid.js";
import "../../Grid/GridItem.js";
const f = `${232 / 16}rem`, h = u(b)`
  width: ${f};
  background: ${({ theme: r }) => r.colors.neutral100};
  position: sticky;
  top: 0;
  height: 100vh;
  overflow-y: auto;
  border-right: 1px solid ${({ theme: r }) => r.colors.neutral200};
  z-index: 1;
`, v = (t) => {
  var e = t, { ariaLabel: r } = e, o = n(e, ["ariaLabel"]);
  return /* @__PURE__ */ d.createElement(h, l({
    "aria-label": r,
    as: "nav"
  }, o));
};
v.propTypes = {
  ariaLabel: c.string.isRequired
};
export {
  v as SubNav,
  f as subNavWidth
};
