// @strapi/design-system/dist/BaseButton/BaseButton.cjs
"use strict";var B=Object.defineProperty;var r=Object.getOwnPropertySymbols;var n=Object.prototype.hasOwnProperty,i=Object.prototype.propertyIsEnumerable;var d=(e,s,t)=>s in e?B(e,s,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[s]=t,c=(e,s)=>{for(var t in s||(s={}))n.call(s,t)&&d(e,t,s[t]);if(r)for(var t of r(s))i.call(s,t)&&d(e,t,s[t]);return e};var p=(e,s)=>{var t={};for(var o in e)n.call(e,o)&&s.indexOf(o)<0&&(t[o]=e[o]);if(e!=null&&r)for(var o of r(e))s.indexOf(o)<0&&i.call(e,o)&&(t[o]=e[o]);return t};Object.defineProperties(exports,{__esModule:{value:!0},[Symbol.toStringTag]:{value:"Module"}});const g=require("react"),_=require("prop-types"),$=require("styled-components"),h=require("../themes/utils.cjs"),u=e=>e&&typeof e=="object"&&"default"in e?e:{default:e},f=u(g),b=u(_),m=u($),y=m.default.button`
  display: flex;
  cursor: pointer;
  padding: ${({theme:e})=>e.spaces[2]};
  border-radius: ${({theme:e})=>e.borderRadius};
  background: ${({theme:e})=>e.colors.neutral0};
  border: 1px solid ${({theme:e})=>e.colors.neutral200};
  svg {
    height: ${({theme:e})=>e.spaces[3]};
    width: ${({theme:e})=>e.spaces[3]};
  }
  svg {
    > g,
    path {
      fill: ${({theme:e})=>e.colors.neutral0};
    }
  }
  &[aria-disabled='true'] {
    pointer-events: none;
  }

  ${h.buttonFocusStyle}
`,a=f.default.forwardRef((q,o)=>{var l=q,{disabled:e,children:s}=l,t=p(l,["disabled","children"]);return f.default.createElement(y,c({ref:o,"aria-disabled":e,type:"button",disabled:e},t),s)});a.displayName="BaseButton";a.defaultProps={disabled:!1};a.propTypes={children:b.default.node.isRequired,disabled:b.default.bool};exports.BaseButton=a;exports.BaseButtonWrapper=y;
